// scs-prospect-portal/src/components/prospects/EditableField.tsx
import { Input } from '../ui/input';
import { Textarea } from '../ui/textarea';
import { Label } from '../ui/label';
import { cn } from '../ui/utils';
import { DdMmYyyyDateInput } from '../ui/dd-mm-yyyy-date-input';

/** Label + required asterisk — matches screening registration (`AddProspect.tsx`). */
export function FormFieldLabel({
  htmlFor,
  label,
  required
}: {
  htmlFor?: string;
  label: string;
  required?: boolean;
}) {
  return (
    <label
      htmlFor={htmlFor}
      className="block mb-2"
      style={{
        fontSize: 'var(--text-base)',
        fontWeight: 'var(--font-weight-normal)',
        color: '#111827'
      }}
    >
      {label}
      {required ? <span style={{ color: '#DC2626' }}>*</span> : null}
    </label>
  );
}

interface EditableFieldProps {
  label: string;
  value: string;
  isEditing: boolean;
  onChange: (value: string) => void;
  required?: boolean;
  type?: 'text' | 'email' | 'tel' | 'date';
  className?: string;
  placeholder?: string;
}

export function EditableField({
  label,
  value,
  isEditing,
  onChange,
  required = false,
  type = 'text',
  className = '',
  placeholder
}: EditableFieldProps) {
  const inputId = `prospect-field-${label.replace(/\s+/g, '-').toLowerCase()}`;

  return (
    <div className={className}>
      <FormFieldLabel htmlFor={inputId} label={label} required={required} />
      {type === 'date' ? (
        <DdMmYyyyDateInput
          id={inputId}
          value={value}
          onChange={onChange}
          disabled={!isEditing}
          required={required}
          placeholder={placeholder || 'DD-MM-YYYY'}
          className={cn(!isEditing && 'bg-[#F9FAFB]')}
        />
      ) : (
        <Input
          id={inputId}
          type={type}
          value={value}
          onChange={(e) => onChange(e.target.value)}
          placeholder={placeholder}
          disabled={!isEditing}
          required={required}
          className={cn('w-full', !isEditing && 'bg-[#F9FAFB]')}
        />
      )}
    </div>
  );
}

interface EditableTextAreaProps {
  label: string;
  value: string;
  isEditing: boolean;
  onChange: (value: string) => void;
  required?: boolean;
  rows?: number;
  className?: string;
  placeholder?: string;
}

export function EditableTextArea({
  label,
  value,
  isEditing,
  onChange,
  required = false,
  rows = 3,
  className = '',
  placeholder
}: EditableTextAreaProps) {
  const id = `prospect-ta-${label.replace(/\s+/g, '-').toLowerCase()}`;

  return (
    <div className={className}>
      <FormFieldLabel htmlFor={id} label={label} required={required} />
      <Textarea
        id={id}
        value={value}
        onChange={(e) => onChange(e.target.value)}
        placeholder={placeholder}
        rows={rows}
        disabled={!isEditing}
        required={required}
        className={cn('w-full min-h-[5rem]', !isEditing && 'bg-[#F9FAFB]')}
      />
    </div>
  );
}

type EditableSelectOption = string | { value: string; label: string };

interface EditableSelectProps {
  label: string;
  value: string;
  isEditing: boolean;
  onChange: (value: string) => void;
  options: EditableSelectOption[];
  required?: boolean;
  className?: string;
  /** First empty option label (e.g. &quot;Select Citizenship&quot;). */
  emptyOptionLabel?: string;
}

function optionValue(opt: EditableSelectOption): string {
  return typeof opt === 'string' ? opt : opt.value;
}

function optionLabel(opt: EditableSelectOption): string {
  return typeof opt === 'string' ? opt : opt.label;
}

export function EditableSelect({
  label,
  value,
  isEditing,
  onChange,
  options,
  required = false,
  className = '',
  emptyOptionLabel = 'Select...'
}: EditableSelectProps) {
  const id = `prospect-sel-${label.replace(/\s+/g, '-').toLowerCase()}`;

  return (
    <div className={className}>
      <Label htmlFor={id} className="mb-2 block text-base font-normal text-gray-900">
        {label}
        {required ? <span className="text-red-600">*</span> : null}
      </Label>
      <select
        id={id}
        value={value}
        onChange={(e) => onChange(e.target.value)}
        disabled={!isEditing}
        required={required}
        className={cn(
          'flex h-9 w-full min-w-0 rounded-lg border border-[var(--input-border)] bg-[var(--input-background)] px-3 py-1 text-sm transition-[color,box-shadow] outline-none',
          'focus-visible:ring-[3px] focus-visible:ring-ring/50',
          'disabled:cursor-not-allowed disabled:opacity-50',
          !isEditing && 'bg-[#F9FAFB]'
        )}
        style={{
          color: 'var(--input-text)',
          borderColor: 'var(--input-border)'
        }}
      >
        {!value && <option value="">{emptyOptionLabel}</option>}
        {options.map((option) => {
          const v = optionValue(option);
          const lbl = optionLabel(option);
          return (
            <option key={v} value={v}>
              {lbl}
            </option>
          );
        })}
      </select>
    </div>
  );
}
