// scs-prospect-portal/src/components/prospects/ProspectDetail.tsx
import { useState, useEffect } from 'react';
import { Button } from '../ui/button';
import { Checkbox } from '../ui/checkbox';
import { Label } from '../ui/label';
import { RadioGroup, RadioGroupItem } from '../ui/radio-group';
import { ChevronLeft, Phone, Mail, Calendar, User, Check, Plus, MoreHorizontal, FileText, Clock, AlertCircle, Copy, RefreshCw, Edit2, Trash2, ChevronDown, ChevronUp, Search, Tag } from 'lucide-react';
import { Page } from '../../App';
import { CreateTaskDialog } from './CreateTaskDialog';
import { AddContactLogDialog } from './AddContactLogDialog';
import { AddNoteDialog } from './AddNoteDialog';
import { ManualRouteDialog } from './ManualRouteDialog';
import { AddTagDialog } from './AddTagDialog';
import { EditableField, EditableTextArea, EditableSelect, FormFieldLabel } from './EditableField';
import { PhonePrefixInput } from '../ui/phone-prefix-input';
import { NricRevealInput } from '../ui/nric-reveal-input';
import { FORM_CARD_SECTION_TITLE_CLASS } from '../../lib/formCardSectionTitle';
import { combineDdMmYyyyAndTime } from '../../lib/dateDdMmYyyy';

interface ProspectDetailProps {
  onNavigate: (page: Page) => void;
  prospectId: string;
}

const RELIGION_OPTIONS = [
  'Buddhism',
  'Catholicism',
  'Christianity',
  'Free Thinker',
  'Hinduism',
  'Islam',
  'Sikhism',
  'Taoism',
  'Others',
] as const;

const PREFERRED_LANGUAGE_OPTIONS = [
  'English',
  'Mandarin',
  'Malay',
  'Tamil',
  'Hokkien',
  'Cantonese',
  'Teochew',
  'Others',
] as const;

/** Values align with screening registration `sourceType` (mammobus / HPV / FIT). */
const SCREENING_SOURCE_TYPE_OPTIONS = [
  'Event',
  'Campaign',
  'Referral',
  'Social Media',
  'Website',
  'Walk-in',
  'Other',
] as const;

/** Matches screening registration `reg-subsidies` CHAS options. */
const CHAS_CARD_TYPE_OPTIONS = ['Blue', 'Orange', 'Green', 'Not Applicable'] as const;

// NoteCard Component with expandable content
function NoteCard({ 
  author, 
  role, 
  timestamp, 
  content, 
  fullContent 
}: { 
  author: string; 
  role: string; 
  timestamp: string; 
  content: string; 
  fullContent: string; 
}) {
  const [isExpanded, setIsExpanded] = useState(false);
  
  return (
    <div className="bg-white rounded-lg border p-4" style={{ borderColor: '#E9ECEF' }}>
      <div className="flex items-start justify-between mb-3">
        <div>
          <div className="flex items-center gap-2 mb-1">
            <span style={{ fontSize: 'var(--text-base)', fontWeight: 'var(--font-weight-semibold)', color: '#111827' }}>
              {author}
            </span>
            <span style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#6B7280' }}>
              • {role}
            </span>
          </div>
          <span style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#6B7280' }}>
            {timestamp}
          </span>
        </div>
        <div className="flex items-center gap-2">
          <button className="p-1 hover:bg-gray-100 rounded" title="Edit note">
            <Edit2 className="w-4 h-4" style={{ color: '#6B7280' }} />
          </button>
          <button className="p-1 hover:bg-gray-100 rounded" title="Delete note">
            <Trash2 className="w-4 h-4" style={{ color: '#6B7280' }} />
          </button>
        </div>
      </div>
      
      <p style={{ 
        fontSize: 'var(--text-sm)', 
        fontWeight: 'var(--font-weight-normal)', 
        color: '#374151',
        lineHeight: '1.5'
      }}>
        {isExpanded ? fullContent : content}
      </p>
      
      {fullContent !== content && (
        <button
          onClick={() => setIsExpanded(!isExpanded)}
          className="mt-2 flex items-center gap-1 hover:opacity-70"
          style={{ 
            color: 'var(--primary)', 
            fontSize: 'var(--text-sm)', 
            fontWeight: 'var(--font-weight-medium)' 
          }}
        >
          {isExpanded ? (
            <>
              Show Less <ChevronUp className="w-4 h-4" />
            </>
          ) : (
            <>
              Show More <ChevronDown className="w-4 h-4" />
            </>
          )}
        </button>
      )}
    </div>
  );
}

// Stage checklist labels
function getStageChecklistLabels(stage: string) {
  const labels: Record<string, string[]> = {
    'Enquiring': [
      'Full name and date of birth collected',
      'NRIC / FIN number recorded',
      'Residency status confirmed (SC / PR / Foreigner)',
      'Mobile number and email captured',
      'Age eligibility confirmed (40 years and above)',
      'Not currently pregnant or breastfeeding',
      'No active breast symptoms present',
      'Date of last mammogram established',
      'Healthier SG or CHAS enrolment checked',
      'First-time screener status noted'
    ],
    'Qualified': [
      'All eligibility criteria met and documented',
      'Subsidy pathway confirmed (Healthier SG / CHAS / self-pay)',
      'Preferred screening date and time captured',
      'Mammobus location / event communicated',
      'Pre-screening instructions shared (no deodorant etc.)',
      'What to bring explained (NRIC, two-piece clothing)',
      'COVID vaccination date noted if within 6 weeks',
      'Family history of breast cancer documented',
      'Any implants or prior surgery flagged for radiographer',
      'Consent to data collection obtained (PDPA)'
    ],
    'Booked': [
      'Appointment confirmation sent (SMS / email)',
      'Reminder sent 1 week before appointment',
      'Reminder sent 3 days before appointment',
      'No-show deposit process explained',
      'Transport or logistics support offered if needed',
      'Emergency contact details collected',
      'Day-of checklist shared with participant',
      'Coordinator notified of any special requirements'
    ],
    'Screened': [
      'Attendance confirmed on screening day',
      'Screening completed without issues',
      'Results communication timeline explained',
      'Results sent / received by participant',
      'Abnormal results — referral pathway activated',
      'Normal results — next screening date communicated',
      'Participant satisfaction / feedback collected',
      'Peer referral ask made (refer a friend)',
      'Re-engagement reminder set (12 or 24 months)',
      'Record updated in system'
    ]
  };
  
  return labels[stage] || [];
}

// Mock data - will be replaced with actual data fetching
const mockProspectData: Record<string, any> = {
  'PROS-001234': {
    id: 'PROS-00123',
    name: 'Lee Wei Xiong',
    gender: 'Female',
    age: 69,
    mobile: '9876 5432',
    email: 'email@email.com',
    source: 'Roadshow',
    assignTo: 'Jasmine Lim',
    status: 'Screening',
    tags: ['Screening', 'High Risk'],
    currentStage: 'Enquiring',
    stages: [
      { name: 'Enquiring', completed: true, current: true },
      { name: 'Qualified', completed: false },
      { name: 'Booked', completed: false },
      { name: 'Screened', completed: false }
    ]
  }
};

export function ProspectDetail({ onNavigate, prospectId }: ProspectDetailProps) {
  const [activeTab, setActiveTab] = useState<'overview' | 'details' | 'appointments' | 'notes' | 'medical-history' | 'other-details' | 'screening'>('overview');
  const [activeSection, setActiveSection] = useState<string>('personal');
  const [activeRoutingSection, setActiveRoutingSection] = useState<string>('eligibility');
  const [activeMedicalSection, setActiveMedicalSection] = useState<string>('family-cancer');
  const [activeOtherDetailsSection, setActiveOtherDetailsSection] = useState<string>('other-medical');
  const [isCreateTaskOpen, setIsCreateTaskOpen] = useState(false);
  const [isAddContactLogOpen, setIsAddContactLogOpen] = useState(false);
  const [isAddNoteOpen, setIsAddNoteOpen] = useState(false);
  const [isManualRouteOpen, setIsManualRouteOpen] = useState(false);
  const [isAddTagOpen, setIsAddTagOpen] = useState(false);
  const [selectedStage, setSelectedStage] = useState('Enquiring');
  const [stageChecklists, setStageChecklists] = useState<Record<string, boolean[]>>({
    'Enquiring': [true, true, true, true, true, false, false, false, false, false],
    'Qualified': [false, false, false, false, false, false, false, false, false, false],
    'Booked': [false, false, false, false, false, false, false, false],
    'Screened': [false, false, false, false, false, false, false, false, false, false]
  });
  
  // Edit mode states
  const [isEditingDetails, setIsEditingDetails] = useState(false);
  const [isEditingMedicalHistory, setIsEditingMedicalHistory] = useState(false);
  const [isEditingOtherDetails, setIsEditingOtherDetails] = useState(false);
  
  // Form data states
  const [detailsFormData, setDetailsFormData] = useState({
    name: 'Lee Wei Xiong',
    nric: 'S1234567D',
    gender: 'Female',
    dateOfBirth: '1956-03-15',
    age: '69',
    mobile: '9876 5432',
    email: 'email@email.com',
    residentialStatus: 'Citizen',
    religion: 'Buddhism',
    preferredLanguages: ['English', 'Mandarin'] as string[],
    race: 'Chinese',
    address: '123 Bedok North Street 1, #05-123, Singapore 460123',
    postalCode: '460123',
    sourceType: 'Event',
    sourceName: 'Community Health Roadshow - Bedok',
    referredBy: 'Dr. Sarah Tan',
    programEnrolled: 'Mammobus',
    chasCardType: 'Blue',
    healthierSg: 'yes',
    firstMammogramScreening: 'no',
    lastScreeningYear: '2023',
    subsidyType: 'Healthier SG',
    hasChronicConditions: 'Yes',
    chronicConditions: 'Hypertension, Diabetes',
    currentMedications: 'Metformin, Lisinopril',
    allergies: 'Penicillin',
    smokingStatus: 'Non-smoker',
    alcoholConsumption: 'Occasional',
    riskLevel: 'High',
    personalCancerHistory: 'No',
    preExistingConditions: 'None',
    familyHistory: 'Mother had breast cancer at age 55',
    cancerScreeningEligibilityCheck: 'No',
    riskFactors: 'Family history of breast cancer, Age above 60',
    pdpaConsent: 'Consented',
    edmSubscription: 'Subscribed',
    consentContact: 'Consented to be contacted for screening appointment booking',
    consentDate: '2025-11-01',
    marketingConsent: 'No',
    /** Screening registration `Appointment Preferences` — same keys as mammobus/hpv/fit `reg-*-appointment` */
    preferredScreeningDate: '15-11-2025',
    preferredTimeSlot: 'morning',
    screeningLocationEvent: 'Bedok Community Centre — Mammobus',
    reviewPeriod: '6months',
    nextReviewDate: '09/10/2026',
  });
  
  const [medicalHistoryFormData, setMedicalHistoryFormData] = useState({
    familyCancerHistory: 'Yes',
    relativesWithCancer: 'Mother (Breast Cancer), Aunt (Ovarian Cancer)',
    personalCancerHistory: 'No',
    previousScreenings: 'Mammogram (2023), Pap Smear (2024)',
    abnormalResults: 'No',
    treatmentHistory: 'None',
    breastSurgery: 'No',
    breastSymptoms: 'No',
    lastMammogramDate: '2023-06-15',
    pregnancyStatus: 'Not applicable',
    breastfeeding: 'No',
    hormonalTherapy: 'No',
    implants: 'No',
    historyOfCancer: 'No',
    diagnosedYear: '-',
    cancerDetail: '-',
    followUpAt: '-',
    surgery: 'No',
    radiationTherapy: 'No',
    chemotherapy: 'No'
  });
  
  const [otherDetailsFormData, setOtherDetailsFormData] = useState({
    emergencyContactName: 'John Doe',
    emergencyContactRelation: 'Spouse',
    emergencyContactPhone: '9123 4567',
    preferredContactTime: 'Morning (9am - 12pm)',
    transportNeeded: 'No',
    languageSupport: 'No',
    accessibilityNeeds: 'Wheelchair access required',
    dietaryRestrictions: 'None',
    religiousConsiderations: 'None',
    insurance: 'Yes',
    insuranceProvider: 'AIA',
    policyNumber: 'AIA123456789',
    employmentStatus: 'Retired',
    education: 'Secondary',
    householdIncome: '$2,000 - $4,000',
    otherMedicalIllness: '1) Left breast abscess surgery, drainage done 3x in 2010 @NCC\n2) Depression, OCD (cu IHH, on Fluoxetine)\n3) L5CS x1 (2006)\n4) L3, L4, L5, S1 bone degeneration, spinal stenosis admitted to NUH in 2018',
    currentMedication: 'Fluoxetine',
    drugAllergy: 'No',
    immunosuppressiveCondition: 'No',
    ageFirstIntercourse: '16',
    multipleSexualPartners: 'Yes',
    smoking: 'Yes',
    smokingDuration: '30',
    sexuallyTransmittedInfections: 'No',
    ocpUse: '0',
    childrenCount: '1',
    hpvVaccinated: 'No',
    cervicalOtherDetails: 'Started smoking at age 24 years old, currently smokes about 8 cig/day. Advised to stop smoking.',
    lastPapTestDate: '4/2/21',
    lastPapTestResult: 'Negative',
    lastMammogramDate: '6/4/26',
    chasCardHolder: 'Yes',
    ageAtMenarche: '12',
    ageAtMenopause: '48',
    ageAtFirstChildbirth: '36',
    hrtDuration: '',
    preMalignantConditions: 'No',
    breastConditionsDetails: 'Breast abscesses x3 in 2012, drainage op done. Now ok'
  });
  
  const [prospectTags, setProspectTags] = useState<Array<{
    id: number;
    name: string;
    category: string;
    color: string;
    textColor: string;
    borderColor: string;
  }>>([
    {
      id: 5,
      name: 'Screening',
      category: 'Screening Stage',
      color: '#F3F4F6',
      textColor: '#374151',
      borderColor: '#E5E7EB'
    },
    {
      id: 3,
      name: 'High Risk',
      category: 'Risk',
      color: '#FEE2E2',
      textColor: '#DC2626',
      borderColor: '#FECACA'
    }
  ]);
  const [tasks, setTasks] = useState<Array<{
    id: string;
    name: string;
    dueDate: string;
    dueTime: string;
    status: 'overdue' | 'upcoming';
    completed: boolean;
  }>>([
    {
      id: '1',
      name: 'Follow up on screening appointment',
      dueDate: 'Nov 18, 2025',
      dueTime: '10:00 AM',
      status: 'overdue',
      completed: false
    },
    {
      id: '2',
      name: 'Send reminder email for health assessment',
      dueDate: 'Nov 22, 2025',
      dueTime: '2:00 PM',
      status: 'upcoming',
      completed: false
    },
    {
      id: '3',
      name: 'Review client documentation',
      dueDate: 'Nov 25, 2025',
      dueTime: '11:00 AM',
      status: 'upcoming',
      completed: false
    }
  ]);
  const [contactLogs, setContactLogs] = useState<Array<{
    id: string;
    contactType: string;
    date: string;
    time: string;
    duration: string;
    recordedBy: string;
    outcome: string;
    message: string;
    notes: string;
  }>>([
    {
      id: '1',
      contactType: 'Phone Call',
      date: 'Nov 15, 2025',
      time: '10:30 AM',
      duration: '15',
      recordedBy: 'Jasmine Lim',
      outcome: 'Successful Contact',
      message: 'Follow-up call regarding screening appointment',
      notes: 'Client confirmed availability for next week. Expressed interest in learning more about our cancer screening programs.'
    },
    {
      id: '2',
      contactType: 'Email',
      date: 'Nov 10, 2025',
      time: '2:15 PM',
      duration: '5',
      recordedBy: 'Sarah Tan',
      outcome: 'Appointment Scheduled',
      message: 'Initial outreach email sent',
      notes: 'Sent detailed information about screening services and available appointment slots.'
    },
    {
      id: '3',
      contactType: 'In-Person Meeting',
      date: 'Nov 5, 2025',
      time: '11:00 AM',
      duration: '30',
      recordedBy: 'Michael Wong',
      outcome: 'Follow-up Required',
      message: 'Roadshow interaction at Tampines Hub',
      notes: 'Met client at community roadshow. Discussed health concerns and provided educational materials. Client requested callback for appointment scheduling.'
    }
  ]);
  
  // Get prospect data or default
  const prospect = mockProspectData[prospectId] || mockProspectData['PROS-001234'];

  // Handle task creation
  const handleCreateTask = (taskData: { name: string; dueDate: string; dueTime: string }) => {
    const combined = combineDdMmYyyyAndTime(taskData.dueDate, taskData.dueTime);
    const when = combined ?? new Date();
    const newTask = {
      id: Date.now().toString(),
      name: taskData.name,
      dueDate: when.toLocaleDateString('en-US', {
        month: 'short',
        day: 'numeric',
        year: 'numeric',
      }),
      dueTime: when.toLocaleTimeString('en-US', {
        hour: 'numeric',
        minute: '2-digit',
        hour12: true,
      }),
      status: when < new Date() ? ('overdue' as const) : ('upcoming' as const),
      completed: false,
    };
    setTasks((prev) => [newTask, ...prev]);
  };

  // Handle contact log creation
  const handleAddContactLog = (logData: {
    contactType: string;
    date: string;
    time: string;
    duration: string;
    recordedBy: string;
    outcome: string;
    message: string;
    notes: string;
  }) => {
    const combined = combineDdMmYyyyAndTime(logData.date, logData.time);
    const when = combined ?? new Date();
    const newLog = {
      id: Date.now().toString(),
      contactType: logData.contactType,
      date: when.toLocaleDateString('en-US', {
        month: 'short',
        day: 'numeric',
        year: 'numeric',
      }),
      time: when.toLocaleTimeString('en-US', {
        hour: 'numeric',
        minute: '2-digit',
        hour12: true,
      }),
      duration: logData.duration,
      recordedBy: logData.recordedBy,
      outcome: logData.outcome,
      message: logData.message,
      notes: logData.notes
    };
    setContactLogs(prev => [newLog, ...prev]);
  };

  // Handle edit mode for Details tab
  const handleEditDetails = () => {
    setIsEditingDetails(true);
  };

  const handleSaveDetails = () => {
    setIsEditingDetails(false);
    // Save logic here - would typically call an API
  };

  const handleCancelDetails = () => {
    setIsEditingDetails(false);
    // Reset form data to original values
    setDetailsFormData({
      name: 'Lee Wei Xiong',
      nric: 'S1234567D',
      gender: 'Female',
      dateOfBirth: '1956-03-15',
      age: '69',
      mobile: '9876 5432',
      email: 'email@email.com',
      residentialStatus: 'Citizen',
      religion: 'Buddhism',
      preferredLanguages: ['English', 'Mandarin'],
      race: 'Chinese',
      address: '123 Bedok North Street 1, #05-123, Singapore 460123',
      postalCode: '460123',
      sourceType: 'Event',
      sourceName: 'Community Health Roadshow - Bedok',
      referredBy: 'Dr. Sarah Tan',
      programEnrolled: 'Mammobus',
      chasCardType: 'Blue',
      healthierSg: 'yes',
      firstMammogramScreening: 'no',
      lastScreeningYear: '2023',
      subsidyType: 'Healthier SG',
      hasChronicConditions: 'Yes',
      chronicConditions: 'Hypertension, Diabetes',
      currentMedications: 'Metformin, Lisinopril',
      allergies: 'Penicillin',
      smokingStatus: 'Non-smoker',
      alcoholConsumption: 'Occasional',
      riskLevel: 'High',
      personalCancerHistory: 'No',
      preExistingConditions: 'None',
      familyHistory: 'Mother had breast cancer at age 55',
      cancerScreeningEligibilityCheck: 'No',
      riskFactors: 'Family history of breast cancer, Age above 60',
      pdpaConsent: 'Consented',
      edmSubscription: 'Subscribed',
      consentContact: 'Consented to be contacted for screening appointment booking',
      consentDate: '2025-11-01',
      marketingConsent: 'No',
      preferredScreeningDate: '15-11-2025',
      preferredTimeSlot: 'morning',
      screeningLocationEvent: 'Bedok Community Centre — Mammobus',
      reviewPeriod: '6months',
      nextReviewDate: '09/10/2026',
    });
  };

  // Handle edit mode for Medical History tab
  const handleEditMedicalHistory = () => {
    setIsEditingMedicalHistory(true);
  };

  const handleSaveMedicalHistory = () => {
    setIsEditingMedicalHistory(false);
    // Save logic here
  };

  const handleCancelMedicalHistory = () => {
    setIsEditingMedicalHistory(false);
    // Reset form data
    setMedicalHistoryFormData({
      familyCancerHistory: 'Yes',
      relativesWithCancer: 'Mother (Breast Cancer), Aunt (Ovarian Cancer)',
      personalCancerHistory: 'No',
      previousScreenings: 'Mammogram (2023), Pap Smear (2024)',
      abnormalResults: 'No',
      treatmentHistory: 'None',
      breastSurgery: 'No',
      breastSymptoms: 'No',
      lastMammogramDate: '2023-06-15',
      pregnancyStatus: 'Not applicable',
      breastfeeding: 'No',
      hormonalTherapy: 'No',
      implants: 'No'
    });
  };

  // Handle edit mode for Other Details tab
  const handleEditOtherDetails = () => {
    setIsEditingOtherDetails(true);
  };

  const handleSaveOtherDetails = () => {
    setIsEditingOtherDetails(false);
    // Save logic here
  };

  const handleCancelOtherDetails = () => {
    setIsEditingOtherDetails(false);
    // Reset form data
    setOtherDetailsFormData({
      emergencyContactName: 'John Doe',
      emergencyContactRelation: 'Spouse',
      emergencyContactPhone: '9123 4567',
      preferredContactTime: 'Morning (9am - 12pm)',
      transportNeeded: 'No',
      languageSupport: 'No',
      accessibilityNeeds: 'Wheelchair access required',
      dietaryRestrictions: 'None',
      religiousConsiderations: 'None',
      insurance: 'Yes',
      insuranceProvider: 'AIA',
      policyNumber: 'AIA123456789',
      employmentStatus: 'Retired',
      education: 'Secondary',
      householdIncome: '$2,000 - $4,000',
      otherMedicalIllness: '1) Left breast abscess surgery, drainage done 3x in 2010 @NCC\n2) Depression, OCD (cu IHH, on Fluoxetine)\n3) L5CS x1 (2006)\n4) L3, L4, L5, S1 bone degeneration, spinal stenosis admitted to NUH in 2018',
      currentMedication: 'Fluoxetine',
      drugAllergy: 'No',
      immunosuppressiveCondition: 'No',
      ageFirstIntercourse: '16',
      multipleSexualPartners: 'Yes',
      smoking: 'Yes',
      smokingDuration: '30',
      sexuallyTransmittedInfections: 'No',
      ocpUse: '0',
      childrenCount: '1',
      hpvVaccinated: 'No',
      cervicalOtherDetails: 'Started smoking at age 24 years old, currently smokes about 8 cig/day. Advised to stop smoking.'
    });
  };

  // Handle checklist toggle
  const handleChecklistToggle = (stage: string, index: number) => {
    setStageChecklists(prev => ({
      ...prev,
      [stage]: prev[stage].map((completed, i) => i === index ? !completed : completed)
    }));
  };

  // Get checklist data with completion state
  const getStageChecklistData = (stage: string) => {
    const labels = getStageChecklistLabels(stage);
    const completions = stageChecklists[stage] || [];
    return labels.map((label, index) => ({
      label,
      completed: completions[index] || false
    }));
  };

  // Get task completion stats for current stage
  const getTaskStats = () => {
    const checklistData = getStageChecklistData(prospect.currentStage);
    const completed = checklistData.filter(t => t.completed).length;
    const total = checklistData.length;
    return { completed, total };
  };

  // State for notes
  const [notes, setNotes] = useState<Array<{
    id: string;
    author: string;
    role: string;
    timestamp: string;
    content: string;
  }>>([
    {
      id: '1',
      author: 'Jasmine Lim',
      role: 'Care Coordinator',
      timestamp: 'Nov 15, 2025 at 3:45 PM',
      content: 'Client expressed strong interest in attending screening workshops. Recommend enrolling in upcoming December session at Bedok Community Center.'
    },
    {
      id: '2',
      author: 'Sarah Tan',
      role: 'Outreach Specialist',
      timestamp: 'Nov 10, 2025 at 10:30 AM',
      content: 'Follow-up needed regarding insurance coverage questions. Client mentioned family history of breast cancer - flagging for priority screening.'
    }
  ]);

  // Handle note creation
  const handleAddNote = (noteContent: string) => {
    const newNote = {
      id: Date.now().toString(),
      author: 'Jasmine Lim',
      role: 'Care Coordinator',
      timestamp: new Date().toLocaleDateString('en-US', { 
        month: 'short', 
        day: 'numeric', 
        year: 'numeric' 
      }) + ' at ' + new Date().toLocaleTimeString('en-US', { 
        hour: 'numeric', 
        minute: '2-digit',
        hour12: true 
      }),
      content: noteContent
    };
    setNotes(prev => [newNote, ...prev]);
  };

  const tabs = [
    { id: 'overview', label: 'Overview' },
    { id: 'details', label: 'Personal Details' },
    { id: 'medical-history', label: 'Medical History' },
    { id: 'other-details', label: 'Other Details' },
    { id: 'screening', label: 'Screening' },
    { id: 'appointments', label: 'Appointments' },
    { id: 'notes', label: 'Notes' }
  ];

  // Scroll tracking for table of contents
  useEffect(() => {
    if (activeTab !== 'details') return;

    const sections = [
      { id: 'personal', element: document.getElementById('personal') },
      { id: 'screening', element: document.getElementById('screening') },
      { id: 'appointment', element: document.getElementById('appointment') },
      { id: 'risk', element: document.getElementById('risk') },
      { id: 'status', element: document.getElementById('status') },
      { id: 'engagement', element: document.getElementById('engagement') },
      { id: 'consent', element: document.getElementById('consent') },
    ].filter(s => s.element !== null);

    const observer = new IntersectionObserver(
      (entries) => {
        entries.forEach((entry) => {
          if (entry.isIntersecting && entry.intersectionRatio >= 0.3) {
            setActiveSection(entry.target.id);
          }
        });
      },
      {
        rootMargin: '-100px 0px -60% 0px',
        threshold: [0, 0.3, 0.6, 1]
      }
    );

    sections.forEach(({ element }) => {
      if (element) observer.observe(element);
    });

    return () => {
      sections.forEach(({ element }) => {
        if (element) observer.unobserve(element);
      });
    };
  }, [activeTab]);

  // Scroll tracking for medical history sections
  useEffect(() => {
    if (activeTab !== 'medical-history') return;

    const sections = [
      { id: 'family-cancer', element: document.getElementById('family-cancer') },
      { id: 'history-cancer', element: document.getElementById('history-cancer') },
      { id: 'treatment', element: document.getElementById('treatment') }
    ].filter(s => s.element !== null);

    const observer = new IntersectionObserver(
      (entries) => {
        entries.forEach((entry) => {
          if (entry.isIntersecting && entry.intersectionRatio >= 0.3) {
            setActiveMedicalSection(entry.target.id);
          }
        });
      },
      {
        rootMargin: '-100px 0px -60% 0px',
        threshold: [0, 0.3, 0.6, 1]
      }
    );

    sections.forEach(({ element }) => {
      if (element) observer.observe(element);
    });

    return () => {
      sections.forEach(({ element }) => {
        if (element) observer.unobserve(element);
      });
    };
  }, [activeTab]);

  // Scroll tracking for other details sections
  useEffect(() => {
    if (activeTab !== 'other-details') return;

    const sections = [
      { id: 'other-medical', element: document.getElementById('other-medical') },
      { id: 'cervical-risk', element: document.getElementById('cervical-risk') },
      { id: 'other-info', element: document.getElementById('other-info') },
      { id: 'breast-risk', element: document.getElementById('breast-risk') }
    ].filter(s => s.element !== null);

    const observer = new IntersectionObserver(
      (entries) => {
        entries.forEach((entry) => {
          if (entry.isIntersecting && entry.intersectionRatio >= 0.3) {
            setActiveOtherDetailsSection(entry.target.id);
          }
        });
      },
      {
        rootMargin: '-100px 0px -60% 0px',
        threshold: [0, 0.3, 0.6, 1]
      }
    );

    sections.forEach(({ element }) => {
      if (element) observer.observe(element);
    });

    return () => {
      sections.forEach(({ element }) => {
        if (element) observer.unobserve(element);
      });
    };
  }, [activeTab]);

  return (
    <div className="h-full flex flex-col" style={{ backgroundColor: '#F9FAFB' }}>
      <div className="flex-1 overflow-auto">
        <div className="bg-white">
          {/* Breadcrumb */}
          <div className="px-6 pt-6 pb-4">
            <div className="flex items-center gap-2 text-sm text-gray-600">
              <button onClick={() => onNavigate('all-prospects')} style={{ color: 'var(--primary)' }} className="hover:underline">Prospect Management</button>
              <span>›</span>
              <span className="text-gray-900">{prospect.id}</span>
            </div>
          </div>

          {/* Header with Back Button and Name */}
          <div className="px-6 pb-4">
            <div className="flex items-center gap-3">
              <button
                onClick={() => onNavigate('all-prospects')}
                className="flex items-center hover:opacity-70"
                style={{ color: '#111827' }}
              >
                <ChevronLeft className="w-5 h-5" />
              </button>
              <h1 style={{ fontSize: 'var(--text-h2)', fontWeight: 'var(--font-weight-semibold)', color: '#111827' }}>
                {prospect.name}
              </h1>
            </div>
          </div>

          {/* Key Info Row */}
          <div className="px-6 pb-6 flex items-center gap-3">
            <span style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#6B7280' }}>
              {prospect.gender} • {prospect.age} years • Next Review: {detailsFormData.nextReviewDate || '—'}
            </span>
            {detailsFormData.riskLevel && (
              <span
                className="px-3 py-1 rounded whitespace-nowrap"
                style={{
                  backgroundColor:
                    detailsFormData.riskLevel === 'High'
                      ? '#FEE2E2'
                      : detailsFormData.riskLevel === 'Medium'
                        ? '#FEF3C7'
                        : '#DBEAFE',
                  color:
                    detailsFormData.riskLevel === 'High'
                      ? '#DC2626'
                      : detailsFormData.riskLevel === 'Medium'
                        ? '#D97706'
                        : '#1E40AF',
                  fontSize: 'var(--text-sm)',
                  fontWeight: 'var(--font-weight-normal)',
                }}
              >
                {detailsFormData.riskLevel} Risk
              </span>
            )}
          </div>

          {/* Status Progress Bar */}
          <div className="px-6 pb-6">
            <div className="flex items-center gap-4">
              {/* Progress Steps with Chevron Design */}
              <div className="flex items-center flex-1 gap-6">
                {prospect.stages.map((stage: any, index: number) => (
                  <div key={index} className="flex items-center flex-1">
                    <button 
                      onClick={() => setSelectedStage(stage.name)}
                      className="relative py-2 flex items-center justify-center w-full transition-all hover:opacity-80"
                      style={{
                        backgroundColor: selectedStage === stage.name ? 'var(--primary)' : '#F3F4F6',
                        borderTopLeftRadius: index === 0 ? '4px' : '0',
                        borderBottomLeftRadius: index === 0 ? '4px' : '0',
                        borderTopRightRadius: index === prospect.stages.length - 1 ? '4px' : '0',
                        borderBottomRightRadius: index === prospect.stages.length - 1 ? '4px' : '0',
                        clipPath: index === prospect.stages.length - 1
                          ? 'polygon(0 0, 100% 0, 100% 100%, 0 100%)'
                          : 'polygon(0 0, calc(100% - 12px) 0, 100% 50%, calc(100% - 12px) 100%, 0 100%)',
                        paddingLeft: '24px',
                        paddingRight: '24px',
                        border: 'none',
                        cursor: 'pointer'
                      }}
                    >
                      <div className="flex items-center gap-2 relative z-10">
                        {selectedStage === stage.name && (
                          <Check className="w-4 h-4" style={{ color: 'white' }} />
                        )}
                        <span style={{ 
                          fontSize: 'var(--text-sm)', 
                          fontWeight: 'var(--font-weight-medium)',
                          color: selectedStage === stage.name ? 'white' : '#9CA3AF'
                        }}>
                          {stage.name}
                        </span>
                      </div>
                    </button>
                  </div>
                ))}
              </div>
              
              {/* Update Button */}
              <Button 
                style={{ 
                  backgroundColor: 'var(--primary)',
                  color: 'white',
                  border: 'none'
                }}
                className="hover:opacity-90"
              >
                <RefreshCw className="w-4 h-4 mr-2" />
                Update
              </Button>
            </div>
          </div>

          {/* Tabs */}
          <div className="border-t" style={{ borderColor: '#E5E7EB' }}>
            <div className="flex gap-8 px-6 overflow-x-auto">
              {tabs.map((tab) => (
                <button
                  key={tab.id}
                  onClick={() => setActiveTab(tab.id as any)}
                  className="py-4 border-b-2 transition-colors whitespace-nowrap"
                  style={{
                    borderColor: activeTab === tab.id ? 'var(--primary)' : 'transparent',
                    color: activeTab === tab.id ? 'var(--primary)' : '#6B7280',
                    fontSize: 'var(--text-sm)',
                    fontWeight: 'var(--font-weight-medium)'
                  }}
                >
                  {tab.label}
                </button>
              ))}
            </div>
          </div>
        </div>

        {/* Tab Content */}
        <div className="p-6">
          {activeTab === 'overview' && (
            <div className="space-y-6">
              {/* Activity Timeline */}
              <div className="bg-white rounded-lg border border-gray-200 p-6">
                <div className="flex items-center justify-between mb-3">
                  <h3 style={{ fontSize: 'var(--text-base)', fontWeight: 'var(--font-weight-semibold)' }}>
                    Activity Timeline
                  </h3>
                  <Button variant="ghost" size="sm">
                    View All
                  </Button>
                </div>
                <div className="space-y-4">
                  {/* Timeline Item Example */}
                  <div className="flex gap-4">
                    <div className="flex flex-col items-center">
                      <div className="w-8 h-8 rounded-full flex items-center justify-center" style={{ backgroundColor: '#EFF6FF' }}>
                        <Phone className="w-4 h-4" style={{ color: 'var(--primary)' }} />
                      </div>
                      <div className="w-0.5 flex-1 bg-gray-200 mt-2" />
                    </div>
                    <div className="flex-1 pb-4">
                      <div className="flex items-start justify-between mb-1">
                        <p style={{ fontSize: 'var(--text-base)', fontWeight: 'var(--font-weight-semibold)' }}>
                          Phone Call
                        </p>
                        <span style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#6B7280' }}>
                          2 hours ago
                        </span>
                      </div>
                      <p style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#6B7280' }}>
                        Contacted prospect regarding screening appointment
                      </p>
                      <p style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#6B7280' }}>
                        By: Jasmine Lim
                      </p>
                    </div>
                  </div>

                  {/* More timeline items would go here */}
                  <div className="flex gap-4">
                    <div className="flex flex-col items-center">
                      <div className="w-8 h-8 rounded-full flex items-center justify-center" style={{ backgroundColor: '#FEF3C7' }}>
                        <AlertCircle className="w-4 h-4 text-yellow-600" />
                      </div>
                      <div className="w-0.5 flex-1 bg-gray-200 mt-2" />
                    </div>
                    <div className="flex-1 pb-4">
                      <div className="flex items-start justify-between mb-1">
                        <p style={{ fontSize: 'var(--text-base)', fontWeight: 'var(--font-weight-semibold)' }}>
                          Status Updated
                        </p>
                        <span style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#6B7280' }}>
                          1 day ago
                        </span>
                      </div>
                      <p style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#6B7280' }}>
                        Status changed from "New" to "Contacted"
                      </p>
                      <p style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#6B7280' }}>
                        By: System
                      </p>
                    </div>
                  </div>
                </div>
              </div>

              {/* Stage Checklist */}
              <div className="bg-white rounded-lg border border-gray-200 p-6">
                <div className="flex items-center justify-between mb-4">
                  <h3 style={{ fontSize: 'var(--text-base)', fontWeight: 'var(--font-weight-semibold)', color: 'var(--primary)' }}>
                    {prospect.currentStage} — tasks
                  </h3>
                  <span style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#9CA3AF' }}>
                    {getTaskStats().completed} / {getTaskStats().total} completed
                  </span>
                </div>
                <div className="space-y-3">
                  {getStageChecklistData(prospect.currentStage).map((task: any, index: number) => (
                    <div key={index} className="flex items-start gap-3">
                      <Checkbox 
                        checked={task.completed}
                        onCheckedChange={() => handleChecklistToggle(prospect.currentStage, index)}
                        className="mt-0.5"
                        style={{
                          borderColor: task.completed ? 'var(--primary)' : '#CED4DA'
                        }}
                      />
                      <label 
                        onClick={() => handleChecklistToggle(prospect.currentStage, index)}
                        style={{ 
                          fontSize: 'var(--text-sm)', 
                          fontWeight: 'var(--font-weight-normal)', 
                          color: task.completed ? '#9CA3AF' : '#374151',
                          textDecoration: task.completed ? 'line-through' : 'none',
                          cursor: 'pointer',
                          flex: 1
                        }}
                      >
                        {task.label}
                      </label>
                    </div>
                  ))}
                </div>
              </div>
            </div>
          )}

          {/* Other tab contents will be added later */}
          {activeTab === 'details' && (
            <div className="flex gap-6">
              {/* Left Sidebar Navigation */}
              <div className="w-64 flex-shrink-0">
                <div className="bg-white rounded-lg border border-gray-200 p-4 sticky top-6">
                  <nav className="space-y-1">
                    {[
                      { id: 'personal', label: 'Personal Information' },
                      { id: 'screening', label: 'Healthier SG & Subsidies' },
                      { id: 'appointment', label: 'Appointment Preferences' },
                      { id: 'risk', label: 'Risk Assessment' },
                      { id: 'status', label: 'Screening Status' },
                      { id: 'engagement', label: 'Engagement' },
                      { id: 'consent', label: 'Consent' }
                    ].map((section) => (
                      <a
                        key={section.id}
                        href={`#${section.id}`}
                        className="block px-3 py-2 rounded transition-colors"
                        style={{
                          fontSize: 'var(--text-sm)',
                          fontWeight: activeSection === section.id ? 'var(--font-weight-semibold)' : 'var(--font-weight-normal)',
                          color: activeSection === section.id ? 'var(--primary)' : '#111827',
                          backgroundColor: activeSection === section.id ? '#F3F4F6' : 'transparent'
                        }}
                        onMouseEnter={(e) => {
                          if (activeSection !== section.id) {
                            e.currentTarget.style.backgroundColor = '#F3F4F6';
                          }
                        }}
                        onMouseLeave={(e) => {
                          if (activeSection !== section.id) {
                            e.currentTarget.style.backgroundColor = 'transparent';
                          }
                        }}
                      >
                        {section.label}
                      </a>
                    ))}
                  </nav>
                </div>
              </div>

              {/* Main Content Area */}
              <div className="flex-1 relative">
                {/* Sticky Edit/Save Buttons */}
                <div className="sticky bottom-0 z-10 py-4 flex justify-end gap-3">
                  {!isEditingDetails ? (
                    <Button 
                      variant="outline" 
                      onClick={handleEditDetails}
                      style={{
                        fontSize: 'var(--text-sm)',
                        fontWeight: 'var(--font-weight-medium)'
                      }}
                    >
                      Edit
                    </Button>
                  ) : (
                    <>
                      <Button 
                        variant="outline" 
                        onClick={handleCancelDetails}
                        style={{
                          fontSize: 'var(--text-sm)',
                          fontWeight: 'var(--font-weight-medium)'
                        }}
                      >
                        Cancel
                      </Button>
                      <Button 
                        onClick={handleSaveDetails}
                        style={{
                          fontSize: 'var(--text-sm)',
                          fontWeight: 'var(--font-weight-medium)',
                          backgroundColor: 'var(--primary)',
                          color: '#FFFFFF'
                        }}
                      >
                        Save
                      </Button>
                    </>
                  )}
                </div>

                <div className="space-y-6">
                  {/* Personal Information — same fields & order as screening registration */}
                  <section id="personal" className="bg-white rounded-lg border border-gray-200 p-6">
                    <h2 className={FORM_CARD_SECTION_TITLE_CLASS}>
                      PERSONAL INFORMATION
                    </h2>
                    
                    <div className="grid grid-cols-1 md:grid-cols-2 gap-6">
                      <EditableField
                        label="Name (as per NRIC)"
                        value={detailsFormData.name}
                        isEditing={isEditingDetails}
                        onChange={(value) => setDetailsFormData({ ...detailsFormData, name: value })}
                        required
                        placeholder="Enter full name as in NRIC"
                      />

                      <EditableSelect
                        label="Residential Status"
                        value={detailsFormData.residentialStatus}
                        isEditing={isEditingDetails}
                        onChange={(value) => setDetailsFormData({ ...detailsFormData, residentialStatus: value })}
                        options={['Singapore Citizen', 'Permanent Resident', 'Foreigner']}
                        emptyOptionLabel="Select Residential Status"
                      />

                      <div>
                        <FormFieldLabel htmlFor="prospect-detail-nric" label="NRIC / FIN Number" required />
                        <NricRevealInput
                          id="prospect-detail-nric"
                          value={detailsFormData.nric}
                          onChange={(value) => setDetailsFormData({ ...detailsFormData, nric: value })}
                          disabled={!isEditingDetails}
                          required
                        />
                      </div>

                      <EditableField
                        label="Date of Birth"
                        value="15/03/1988"
                        isEditing={isEditingDetails}
                        onChange={(value) => setDetailsFormData({ ...detailsFormData, dateOfBirth: value })}
                        type="date"
                        required
                      />

                      <EditableSelect
                        label="Gender"
                        value={detailsFormData.gender}
                        isEditing={isEditingDetails}
                        onChange={(value) => setDetailsFormData({ ...detailsFormData, gender: value })}
                        options={['Female', 'Male', 'Other']}
                        required
                        emptyOptionLabel="Select Gender"
                      />

                      <EditableSelect
                        label="Race"
                        value={detailsFormData.race}
                        isEditing={isEditingDetails}
                        onChange={(value) => setDetailsFormData({ ...detailsFormData, race: value })}
                        options={['Chinese', 'Malay', 'Indian', 'Eurasian', 'Others']}
                        emptyOptionLabel="Select Race"
                      />

                      <EditableSelect
                        label="Religion"
                        value={detailsFormData.religion}
                        isEditing={isEditingDetails}
                        onChange={(value) => setDetailsFormData({ ...detailsFormData, religion: value })}
                        options={[...RELIGION_OPTIONS]}
                        emptyOptionLabel="Select Religion"
                      />

                      <div>
                        <FormFieldLabel label="Contact Number" required />
                        <PhonePrefixInput
                          value={detailsFormData.mobile}
                          onChange={(value) => setDetailsFormData({ ...detailsFormData, mobile: value })}
                          disabled={!isEditingDetails}
                          required
                          placeholder="E.g. 8123 4567"
                        />
                      </div>

                      <EditableField
                        label="Email"
                        value={detailsFormData.email}
                        isEditing={isEditingDetails}
                        onChange={(value) => setDetailsFormData({ ...detailsFormData, email: value })}
                        type="email"
                        placeholder="Enter your email"
                      />

                      <div className="md:col-span-2">
                        <FormFieldLabel label="Preferred Language" />
                        {!isEditingDetails ? (
                          <p
                            style={{
                              fontSize: 'var(--text-base)',
                              color: '#111827',
                              margin: 0,
                            }}
                          >
                            {detailsFormData.preferredLanguages.length > 0
                              ? detailsFormData.preferredLanguages.join(', ')
                              : '—'}
                          </p>
                        ) : (
                          <div className="space-y-2">
                            {PREFERRED_LANGUAGE_OPTIONS.map((lang) => (
                              <label
                                key={lang}
                                className="flex items-center gap-2 cursor-pointer"
                              >
                                <Checkbox
                                  checked={detailsFormData.preferredLanguages.includes(lang)}
                                  onCheckedChange={() =>
                                    setDetailsFormData((prev) => ({
                                      ...prev,
                                      preferredLanguages: prev.preferredLanguages.includes(lang)
                                        ? prev.preferredLanguages.filter((l) => l !== lang)
                                        : [...prev.preferredLanguages, lang],
                                    }))
                                  }
                                />
                                <span style={{ fontSize: 'var(--text-base)', color: '#111827' }}>
                                  {lang}
                                </span>
                              </label>
                            ))}
                          </div>
                        )}
                      </div>
                    </div>
                  </section>

                  {/* Healthier SG & Subsidies — maps to screening registration `reg-subsidies` */}
                  <section id="screening" className="bg-white rounded-lg border border-gray-200 p-6">
                    <h2 className={FORM_CARD_SECTION_TITLE_CLASS}>
                      HEALTHIER SG &amp; SUBSIDIES
                    </h2>
                    
                    <div className="grid grid-cols-1 md:grid-cols-2 gap-6">
                      <EditableSelect
                        label="CHAS Card Type"
                        value={detailsFormData.chasCardType}
                        isEditing={isEditingDetails}
                        onChange={(value) => setDetailsFormData({ ...detailsFormData, chasCardType: value })}
                        options={[...CHAS_CARD_TYPE_OPTIONS]}
                        emptyOptionLabel="Select CHAS Card Type"
                      />

                      <div>
                        <FormFieldLabel label="Are you enrolled under Healthier SG?" />
                        <select
                          id="prospect-healthier-sg"
                          value={detailsFormData.healthierSg}
                          onChange={(e) =>
                            setDetailsFormData({ ...detailsFormData, healthierSg: e.target.value })
                          }
                          disabled={!isEditingDetails}
                          className="flex h-9 w-full min-w-0 rounded-lg border border-[var(--input-border)] bg-[var(--input-background)] px-3 py-1 text-sm outline-none focus-visible:ring-[3px] focus-visible:ring-ring/50 disabled:cursor-not-allowed disabled:opacity-50"
                          style={{ color: 'var(--input-text)', borderColor: 'var(--input-border)' }}
                        >
                          <option value="">Select Enrolment Status</option>
                          <option value="yes">Yes</option>
                          <option value="no">No</option>
                          <option value="unsure">Unsure / Prefer not to say</option>
                        </select>
                      </div>

                      <div className="md:col-span-2 space-y-3">
                        <FormFieldLabel label="Is this your first mammogram screening?" />
                        <RadioGroup
                          value={detailsFormData.firstMammogramScreening}
                          onValueChange={(v) =>
                            setDetailsFormData({ ...detailsFormData, firstMammogramScreening: v })
                          }
                          disabled={!isEditingDetails}
                          className="flex flex-wrap gap-6"
                        >
                          <div className="flex items-center gap-2">
                            <RadioGroupItem value="yes" id="first-mammo-yes" />
                            <Label htmlFor="first-mammo-yes" className="cursor-pointer font-normal text-gray-900">
                              Yes
                            </Label>
                          </div>
                          <div className="flex items-center gap-2">
                            <RadioGroupItem value="no" id="first-mammo-no" />
                            <Label htmlFor="first-mammo-no" className="cursor-pointer font-normal text-gray-900">
                              No
                            </Label>
                          </div>
                        </RadioGroup>
                      </div>

                      <EditableField
                        label="Year of Last Screening"
                        value={detailsFormData.lastScreeningYear}
                        isEditing={isEditingDetails}
                        onChange={(value) =>
                          setDetailsFormData({ ...detailsFormData, lastScreeningYear: value })
                        }
                        placeholder="Enter Year of Last Screening"
                      />
                    </div>
                  </section>

                  {/* Appointment Preferences — maps to screening registration `reg-appointment` / `reg-hpv-appointment` / `reg-fit-appointment` */}
                  <section id="appointment" className="bg-white rounded-lg border border-gray-200 p-6">
                    <h2 className={FORM_CARD_SECTION_TITLE_CLASS}>
                      APPOINTMENT PREFERENCES
                    </h2>
                    <div className="grid grid-cols-1 md:grid-cols-2 gap-6">
                      <EditableField
                        label="Preferred Screening Date"
                        value={detailsFormData.preferredScreeningDate}
                        isEditing={isEditingDetails}
                        onChange={(value) =>
                          setDetailsFormData({ ...detailsFormData, preferredScreeningDate: value })
                        }
                        placeholder="DD-MM-YYYY"
                      />
                      <EditableSelect
                        label="Preferred Time Slot"
                        value={detailsFormData.preferredTimeSlot}
                        isEditing={isEditingDetails}
                        onChange={(value) =>
                          setDetailsFormData({ ...detailsFormData, preferredTimeSlot: value })
                        }
                        options={[
                          { value: 'morning', label: 'Morning' },
                          { value: 'afternoon', label: 'Afternoon' },
                          { value: 'evening', label: 'Evening' },
                        ]}
                        emptyOptionLabel="Select Preferred Time Slot"
                      />
                      <div className="md:col-span-2">
                        <EditableField
                          label="Screening Location / Event"
                          value={detailsFormData.screeningLocationEvent}
                          isEditing={isEditingDetails}
                          onChange={(value) =>
                            setDetailsFormData({ ...detailsFormData, screeningLocationEvent: value })
                          }
                          placeholder="Enter Screening Location / Event"
                        />
                      </div>
                    </div>
                  </section>

                  {/* Risk Assessment Section */}
                  <section id="risk" className="bg-white rounded-lg border border-gray-200 p-6">
                    <h2 className={FORM_CARD_SECTION_TITLE_CLASS}>
                      RISK ASSESSMENT
                    </h2>
                    
                    <div className="flex flex-col gap-6">
                      <EditableSelect
                        label="Risk Level"
                        value={detailsFormData.riskLevel || ''}
                        isEditing={isEditingDetails}
                        onChange={(value) => setDetailsFormData({ ...detailsFormData, riskLevel: value })}
                        options={['Low', 'Medium', 'High']}
                        emptyOptionLabel="Select Risk Level"
                      />

                      <EditableTextArea
                        label="Personal History of Cancer"
                        value={detailsFormData.personalCancerHistory}
                        isEditing={isEditingDetails}
                        onChange={(value) =>
                          setDetailsFormData({ ...detailsFormData, personalCancerHistory: value })
                        }
                        rows={4}
                        placeholder="Enter Personal History of Cancer"
                      />

                      <EditableTextArea
                        label="Pre-existing Health Conditions"
                        value={detailsFormData.preExistingConditions}
                        isEditing={isEditingDetails}
                        onChange={(value) =>
                          setDetailsFormData({ ...detailsFormData, preExistingConditions: value })
                        }
                        rows={4}
                        placeholder="Enter Pre-existing Health Conditions"
                      />

                      <EditableTextArea
                        label="Family History of Cancer"
                        value={detailsFormData.familyHistory}
                        isEditing={isEditingDetails}
                        onChange={(value) => setDetailsFormData({ ...detailsFormData, familyHistory: value })}
                        rows={4}
                        placeholder="Enter Family History of Cancer"
                      />

                      <div className="flex items-start gap-3">
                        <Checkbox
                          id="risk-eligibility-check"
                          checked={detailsFormData.cancerScreeningEligibilityCheck === 'Yes'}
                          onCheckedChange={(c) =>
                            setDetailsFormData({
                              ...detailsFormData,
                              cancerScreeningEligibilityCheck: c === true ? 'Yes' : 'No',
                            })
                          }
                          disabled={!isEditingDetails}
                        />
                        <label
                          htmlFor="risk-eligibility-check"
                          className="text-base text-gray-900 cursor-pointer pt-0.5"
                          style={{ fontSize: 'var(--text-base)' }}
                        >
                          Cancer Screening Eligibility Check
                        </label>
                      </div>
                    </div>
                  </section>

                  {/* Screening Status Section */}
                  <section id="status" className="bg-white rounded-lg border border-gray-200 p-6">
                    <h2 className={FORM_CARD_SECTION_TITLE_CLASS}>
                      SCREENING STATUS
                    </h2>
                    
                    <div className="space-y-4">
                      <EditableTextArea
                        label="Screening Eligible For & Signed Up"
                        value="Mammogram (Mammobus) - Booked&#10;HPV Screening - Interested"
                        isEditing={isEditingDetails}
                        onChange={(value) => {}}
                        rows={3}
                      />

                      <EditableTextArea
                        label="Follow-up Notes for CN"
                        value="Booked for Mammobus on 15 Nov 2025 at Bedok CC. Follow up on HPV screening interest after mammogram results."
                        isEditing={isEditingDetails}
                        onChange={(value) => {}}
                        rows={3}
                      />
                    </div>
                  </section>

                  {/* Engagement — maps to screening registration Engagement section */}
                  <section id="engagement" className="bg-white rounded-lg border border-gray-200 p-6">
                    <h2 className={FORM_CARD_SECTION_TITLE_CLASS}>
                      ENGAGEMENT
                    </h2>
                    
                    <div className="grid grid-cols-1 md:grid-cols-2 gap-6">
                      <EditableSelect
                        label="How did you hear about us?"
                        value={detailsFormData.sourceType}
                        isEditing={isEditingDetails}
                        onChange={(value) => setDetailsFormData({ ...detailsFormData, sourceType: value })}
                        options={[...SCREENING_SOURCE_TYPE_OPTIONS]}
                        emptyOptionLabel="Select Source Type"
                      />
                      <EditableField
                        label="Campaign / Event Name"
                        value={detailsFormData.sourceName}
                        isEditing={isEditingDetails}
                        onChange={(value) => setDetailsFormData({ ...detailsFormData, sourceName: value })}
                        placeholder="e.g. Pink for Life 2025"
                      />
                    </div>
                  </section>

                  {/* Consent — maps to screening registration Consent section */}
                  <section id="consent" className="bg-white rounded-lg border border-gray-200 p-6">
                    <h2 className={FORM_CARD_SECTION_TITLE_CLASS}>
                      CONSENT
                    </h2>
                    
                    <div className="grid grid-cols-1 md:grid-cols-2 gap-6">
                      <EditableField
                        label="PDPA Consent"
                        value={detailsFormData.pdpaConsent}
                        isEditing={isEditingDetails}
                        onChange={(value) => setDetailsFormData({ ...detailsFormData, pdpaConsent: value })}
                      />

                      <EditableField
                        label="eDM Subscription"
                        value={detailsFormData.edmSubscription}
                        isEditing={isEditingDetails}
                        onChange={(value) => setDetailsFormData({ ...detailsFormData, edmSubscription: value })}
                      />

                      <EditableTextArea
                        label="Consent for SCS to Contact"
                        value={detailsFormData.consentContact}
                        isEditing={isEditingDetails}
                        onChange={(value) => setDetailsFormData({ ...detailsFormData, consentContact: value })}
                        rows={3}
                        className="md:col-span-2"
                      />
                    </div>
                  </section>
                </div>
              </div>
            </div>
          )}

          {/* Appointments Tab */}
          {activeTab === 'appointments' && (
            <div className="space-y-6">
              {/* Search Bar and Create Button */}
              <div className="flex items-center gap-4">
                <div className="flex-1 relative">
                  <input
                    type="text"
                    placeholder="Search appointments..."
                    className="w-full px-3 py-2 rounded border"
                    style={{
                      backgroundColor: '#FFFFFF',
                      borderColor: '#CED4DA',
                      fontSize: 'var(--text-sm)',
                      fontWeight: 'var(--font-weight-normal)',
                      color: '#272B30'
                    }}
                  />
                </div>
                <Button 
                  style={{ 
                    backgroundColor: 'var(--primary)',
                    color: 'white',
                    border: 'none'
                  }}
                  className="hover:opacity-90"
                >
                  <Plus className="w-4 h-4 mr-2" />
                  Create Appointment
                </Button>
              </div>

              {/* Appointment List */}
              <div className="space-y-4">
                {/* Appointment 1 */}
                <div className="bg-white rounded-lg border border-gray-200 p-6 flex gap-6">
                  {/* Date Section */}
                  <div className="flex flex-col items-center justify-center" style={{ minWidth: '100px' }}>
                    <div style={{ 
                      fontSize: 'var(--text-xs)', 
                      fontWeight: 'var(--font-weight-medium)',
                      color: '#6B7280',
                      marginBottom: '4px'
                    }}>
                      Thu
                    </div>
                    <div style={{ 
                      fontSize: 'var(--text-h2)', 
                      fontWeight: 'var(--font-weight-semibold)',
                      color: '#111827',
                      lineHeight: '1'
                    }}>
                      20
                    </div>
                    <div style={{ 
                      fontSize: 'var(--text-xs)', 
                      fontWeight: 'var(--font-weight-medium)',
                      color: '#6B7280',
                      marginTop: '4px'
                    }}>
                      Aug 2025
                    </div>
                  </div>

                  {/* Appointment Details */}
                  <div className="flex-1 border-l border-gray-200 pl-6">
                    <div className="flex items-start justify-between mb-2">
                      <h3 style={{ 
                        fontSize: 'var(--text-base)', 
                        fontWeight: 'var(--font-weight-semibold)',
                        color: '#111827'
                      }}>
                        Screening
                      </h3>
                      <span
                        className="px-3 py-1 rounded"
                        style={{
                          backgroundColor: '#DBEAFE',
                          color: '#1E40AF',
                          fontSize: 'var(--text-sm)',
                          fontWeight: 'var(--font-weight-normal)'
                        }}
                      >
                        Rescheduled
                      </span>
                    </div>
                    <div className="flex items-center gap-2 mb-3">
                      <Clock className="w-4 h-4" style={{ color: '#6B7280' }} />
                      <span style={{ 
                        fontSize: 'var(--text-sm)', 
                        fontWeight: 'var(--font-weight-normal)',
                        color: '#6B7280'
                      }}>
                        08:30 AM - 09:00 AM
                      </span>
                    </div>
                    <div className="flex items-center gap-2 mb-3">
                      <User className="w-4 h-4" style={{ color: '#6B7280' }} />
                      <span style={{ 
                        fontSize: 'var(--text-sm)', 
                        fontWeight: 'var(--font-weight-normal)',
                        color: '#6B7280'
                      }}>
                        Dr. Sarah Tan
                      </span>
                    </div>
                    <p style={{ 
                      fontSize: 'var(--text-sm)', 
                      fontWeight: 'var(--font-weight-normal)',
                      color: '#6B7280'
                    }}>
                      Location: Singapore Cancer Centre, Level 3
                    </p>
                  </div>
                </div>

                {/* Appointment 2 */}
                <div className="bg-white rounded-lg border border-gray-200 p-6 flex gap-6">
                  {/* Date Section */}
                  <div className="flex flex-col items-center justify-center" style={{ minWidth: '100px' }}>
                    <div style={{ 
                      fontSize: 'var(--text-xs)', 
                      fontWeight: 'var(--font-weight-medium)',
                      color: '#6B7280',
                      marginBottom: '4px'
                    }}>
                      Mon
                    </div>
                    <div style={{ 
                      fontSize: 'var(--text-h2)', 
                      fontWeight: 'var(--font-weight-semibold)',
                      color: '#111827',
                      lineHeight: '1'
                    }}>
                      15
                    </div>
                    <div style={{ 
                      fontSize: 'var(--text-xs)', 
                      fontWeight: 'var(--font-weight-medium)',
                      color: '#6B7280',
                      marginTop: '4px'
                    }}>
                      Jul 2025
                    </div>
                  </div>

                  {/* Appointment Details */}
                  <div className="flex-1 border-l border-gray-200 pl-6">
                    <div className="flex items-start justify-between mb-2">
                      <h3 style={{ 
                        fontSize: 'var(--text-base)', 
                        fontWeight: 'var(--font-weight-semibold)',
                        color: '#111827'
                      }}>
                        Screening
                      </h3>
                      <span
                        className="px-3 py-1 rounded"
                        style={{
                          backgroundColor: '#D1FAE5',
                          color: '#065F46',
                          fontSize: 'var(--text-sm)',
                          fontWeight: 'var(--font-weight-normal)'
                        }}
                      >
                        Completed
                      </span>
                    </div>
                    <div className="flex items-center gap-2 mb-3">
                      <Clock className="w-4 h-4" style={{ color: '#6B7280' }} />
                      <span style={{ 
                        fontSize: 'var(--text-sm)', 
                        fontWeight: 'var(--font-weight-normal)',
                        color: '#6B7280'
                      }}>
                        10:00 AM - 10:30 AM
                      </span>
                    </div>
                    <div className="flex items-center gap-2 mb-3">
                      <User className="w-4 h-4" style={{ color: '#6B7280' }} />
                      <span style={{ 
                        fontSize: 'var(--text-sm)', 
                        fontWeight: 'var(--font-weight-normal)',
                        color: '#6B7280'
                      }}>
                        Dr. Michael Wong
                      </span>
                    </div>
                    <p style={{ 
                      fontSize: 'var(--text-sm)', 
                      fontWeight: 'var(--font-weight-normal)',
                      color: '#6B7280'
                    }}>
                      Location: Singapore Cancer Centre, Level 2
                    </p>
                  </div>
                </div>

                {/* Appointment 3 */}
                <div className="bg-white rounded-lg border border-gray-200 p-6 flex gap-6">
                  {/* Date Section */}
                  <div className="flex flex-col items-center justify-center" style={{ minWidth: '100px' }}>
                    <div style={{ 
                      fontSize: 'var(--text-xs)', 
                      fontWeight: 'var(--font-weight-medium)',
                      color: '#6B7280',
                      marginBottom: '4px'
                    }}>
                      Fri
                    </div>
                    <div style={{ 
                      fontSize: 'var(--text-h2)', 
                      fontWeight: 'var(--font-weight-semibold)',
                      color: '#111827',
                      lineHeight: '1'
                    }}>
                      10
                    </div>
                    <div style={{ 
                      fontSize: 'var(--text-xs)', 
                      fontWeight: 'var(--font-weight-medium)',
                      color: '#6B7280',
                      marginTop: '4px'
                    }}>
                      Jun 2025
                    </div>
                  </div>

                  {/* Appointment Details */}
                  <div className="flex-1 border-l border-gray-200 pl-6">
                    <div className="flex items-start justify-between mb-2">
                      <h3 style={{ 
                        fontSize: 'var(--text-base)', 
                        fontWeight: 'var(--font-weight-semibold)',
                        color: '#111827'
                      }}>
                        Screening
                      </h3>
                      <span
                        className="px-3 py-1 rounded"
                        style={{
                          backgroundColor: '#D1FAE5',
                          color: '#065F46',
                          fontSize: 'var(--text-sm)',
                          fontWeight: 'var(--font-weight-normal)'
                        }}
                      >
                        Completed
                      </span>
                    </div>
                    <div className="flex items-center gap-2 mb-3">
                      <Clock className="w-4 h-4" style={{ color: '#6B7280' }} />
                      <span style={{ 
                        fontSize: 'var(--text-sm)', 
                        fontWeight: 'var(--font-weight-normal)',
                        color: '#6B7280'
                      }}>
                        02:00 PM - 02:30 PM
                      </span>
                    </div>
                    <div className="flex items-center gap-2 mb-3">
                      <User className="w-4 h-4" style={{ color: '#6B7280' }} />
                      <span style={{ 
                        fontSize: 'var(--text-sm)', 
                        fontWeight: 'var(--font-weight-normal)',
                        color: '#6B7280'
                      }}>
                        Dr. Sarah Tan
                      </span>
                    </div>
                    <p style={{ 
                      fontSize: 'var(--text-sm)', 
                      fontWeight: 'var(--font-weight-normal)',
                      color: '#6B7280'
                    }}>
                      Location: Singapore Cancer Centre, Level 3
                    </p>
                  </div>
                </div>
              </div>
            </div>
          )}

          {/* Notes Tab */}
          {activeTab === 'notes' && (
            <div className="space-y-6">
              {/* Header with Last Updated and Add Notes */}
              <div className="flex items-center justify-between">
                <div className="flex items-center gap-2">
                  <RefreshCw className="w-4 h-4" style={{ color: '#6B7280' }} />
                  <span style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#6B7280' }}>
                    Last updated on 17 July 2025 by John Smith
                  </span>
                </div>
                <Button
                  variant="outline"
                  onClick={() => setIsAddNoteOpen(true)}
                  style={{
                    borderColor: 'var(--primary)',
                    color: 'var(--primary)',
                    backgroundColor: 'white',
                    fontSize: 'var(--text-sm)',
                    fontWeight: 'var(--font-weight-medium)'
                  }}
                  className="hover:bg-purple-50"
                >
                  <Plus className="w-4 h-4 mr-2" />
                  Add Notes
                </Button>
              </div>

              {/* Discussion Notes Section */}
              <div>
                <div className="space-y-4">
                  {notes.map((note) => (
                    <NoteCard
                      key={note.id}
                      author={note.author}
                      role={note.role}
                      timestamp={note.timestamp}
                      content={note.content}
                      fullContent={note.content}
                    />
                  ))}
                </div>
              </div>

              {/* Pagination */}
              <div className="flex items-center justify-center gap-2 mt-6">
                <button
                  className="w-8 h-8 rounded border flex items-center justify-center hover:bg-gray-50"
                  style={{
                    borderColor: '#DEE2E6',
                    color: '#6B7280'
                  }}
                >
                  <ChevronLeft className="w-4 h-4" />
                </button>
                <button
                  className="w-8 h-8 rounded flex items-center justify-center"
                  style={{
                    backgroundColor: 'var(--primary)',
                    color: 'white',
                    fontSize: 'var(--text-sm)',
                    fontWeight: 'var(--font-weight-medium)'
                  }}
                >
                  1
                </button>
                <button
                  className="w-8 h-8 rounded border flex items-center justify-center hover:bg-gray-50"
                  style={{
                    borderColor: '#DEE2E6',
                    color: '#374151',
                    fontSize: 'var(--text-sm)',
                    fontWeight: 'var(--font-weight-medium)'
                  }}
                >
                  2
                </button>
                <span style={{ color: '#6B7280', fontSize: 'var(--text-sm)' }}>...</span>
                <button
                  className="w-8 h-8 rounded border flex items-center justify-center hover:bg-gray-50"
                  style={{
                    borderColor: '#DEE2E6',
                    color: '#6B7280'
                  }}
                >
                  <ChevronLeft className="w-4 h-4 rotate-180" />
                </button>
              </div>
            </div>
          )}

          {/* Medical History Tab */}
          {activeTab === 'medical-history' && (
            <div className="flex gap-6">
              {/* Left Sidebar Navigation */}
              <div className="w-64 flex-shrink-0">
                <div className="bg-white rounded-lg border border-gray-200 p-4 sticky top-6">
                  <nav className="space-y-1">
                      {[
                        { id: 'family-cancer', label: 'Family Cancer Background' },
                        { id: 'history-cancer', label: 'History of Cancer' },
                        { id: 'treatment', label: 'Treatment Done' }
                      ].map((section) => (
                        <a
                          key={section.id}
                          href={`#${section.id}`}
                          className="block px-3 py-2 rounded transition-colors"
                          style={{
                            fontSize: 'var(--text-sm)',
                            fontWeight: activeMedicalSection === section.id ? 'var(--font-weight-semibold)' : 'var(--font-weight-normal)',
                            color: activeMedicalSection === section.id ? 'var(--primary)' : '#111827',
                            backgroundColor: activeMedicalSection === section.id ? '#F3F4F6' : 'transparent'
                          }}
                          onMouseEnter={(e) => {
                            if (activeMedicalSection !== section.id) {
                              e.currentTarget.style.backgroundColor = '#F3F4F6';
                            }
                          }}
                          onMouseLeave={(e) => {
                            if (activeMedicalSection !== section.id) {
                              e.currentTarget.style.backgroundColor = 'transparent';
                            }
                          }}
                        >
                          {section.label}
                        </a>
                      ))}
                    </nav>
                  </div>
                </div>

                {/* Main Content Area */}
                <div className="flex-1 relative">
                  {/* Sticky Edit/Save Buttons */}
                  <div className="sticky bottom-0 z-10 py-4 flex justify-end gap-3">
                    {!isEditingMedicalHistory ? (
                      <Button 
                        variant="outline" 
                        onClick={handleEditMedicalHistory}
                        style={{
                          fontSize: 'var(--text-sm)',
                          fontWeight: 'var(--font-weight-medium)'
                        }}
                      >
                        Edit
                      </Button>
                    ) : (
                      <>
                        <Button 
                          variant="outline" 
                          onClick={handleCancelMedicalHistory}
                          style={{
                            fontSize: 'var(--text-sm)',
                            fontWeight: 'var(--font-weight-medium)'
                          }}
                        >
                          Cancel
                        </Button>
                        <Button 
                          onClick={handleSaveMedicalHistory}
                          style={{
                            fontSize: 'var(--text-sm)',
                            fontWeight: 'var(--font-weight-medium)',
                            backgroundColor: 'var(--primary)',
                            color: '#FFFFFF'
                          }}
                        >
                          Save
                        </Button>
                      </>
                    )}
                  </div>

                  <div className="space-y-6">
                  {/* Family Cancer Background Section */}
                  <section id="family-cancer" className="bg-white rounded-lg border border-gray-200 p-6">
                    <h2 className={FORM_CARD_SECTION_TITLE_CLASS}>
                      FAMILY CANCER BACKGROUND
                    </h2>
                    
                    <div className="grid grid-cols-2 gap-6">
                      <EditableSelect
                        label="Breast Cancer"
                        value="Yes"
                        isEditing={isEditingMedicalHistory}
                        onChange={(value) => {}}
                        options={['Yes', 'No']}
                      />
                      
                      <EditableSelect
                        label="Colorectal Cancer"
                        value="No"
                        isEditing={isEditingMedicalHistory}
                        onChange={(value) => {}}
                        options={['Yes', 'No']}
                      />
                      
                      <EditableSelect
                        label="Ovarian Cancer"
                        value="No"
                        isEditing={isEditingMedicalHistory}
                        onChange={(value) => {}}
                        options={['Yes', 'No']}
                      />
                      
                      <EditableSelect
                        label="Other Cancer"
                        value="No"
                        isEditing={isEditingMedicalHistory}
                        onChange={(value) => {}}
                        options={['Yes', 'No']}
                      />
                    </div>
                  </section>

                  {/* History of Cancer Section */}
                  <section id="history-cancer" className="bg-white rounded-lg border border-gray-200 p-6">
                    <h2 className={FORM_CARD_SECTION_TITLE_CLASS}>
                      HISTORY OF CANCER
                    </h2>
                    
                    <div className="grid grid-cols-2 gap-6">
                      <EditableSelect
                        label="History of Cancer"
                        value={medicalHistoryFormData.historyOfCancer}
                        isEditing={isEditingMedicalHistory}
                        onChange={(value) => setMedicalHistoryFormData({...medicalHistoryFormData, historyOfCancer: value})}
                        options={['Yes', 'No']}
                      />
                      
                      <EditableField
                        label="Diagnosed Year"
                        value={medicalHistoryFormData.diagnosedYear}
                        isEditing={isEditingMedicalHistory}
                        onChange={(value) => setMedicalHistoryFormData({...medicalHistoryFormData, diagnosedYear: value})}
                      />
                      
                      <EditableTextArea
                        label="Cancer Detail"
                        value={medicalHistoryFormData.cancerDetail}
                        isEditing={isEditingMedicalHistory}
                        onChange={(value) => setMedicalHistoryFormData({...medicalHistoryFormData, cancerDetail: value})}
                        className="col-span-2"
                        rows={3}
                      />
                      
                      <EditableField
                        label="Follow Up At"
                        value={medicalHistoryFormData.followUpAt}
                        isEditing={isEditingMedicalHistory}
                        onChange={(value) => setMedicalHistoryFormData({...medicalHistoryFormData, followUpAt: value})}
                        className="col-span-2"
                      />
                    </div>
                  </section>

                  {/* Treatment Done Section */}
                  <section id="treatment" className="bg-white rounded-lg border border-gray-200 p-6">
                    <h2 className={FORM_CARD_SECTION_TITLE_CLASS}>
                      TREATMENT DONE
                    </h2>
                    
                    <div className="grid grid-cols-2 gap-6">
                      <EditableSelect
                        label="Surgery"
                        value={medicalHistoryFormData.surgery}
                        isEditing={isEditingMedicalHistory}
                        onChange={(value) => setMedicalHistoryFormData({...medicalHistoryFormData, surgery: value})}
                        options={['Yes', 'No']}
                      />
                      
                      <EditableSelect
                        label="Radiation Therapy"
                        value={medicalHistoryFormData.radiationTherapy}
                        isEditing={isEditingMedicalHistory}
                        onChange={(value) => setMedicalHistoryFormData({...medicalHistoryFormData, radiationTherapy: value})}
                        options={['Yes', 'No']}
                      />
                      
                      <EditableSelect
                        label="Chemotherapy"
                        value={medicalHistoryFormData.chemotherapy}
                        isEditing={isEditingMedicalHistory}
                        onChange={(value) => setMedicalHistoryFormData({...medicalHistoryFormData, chemotherapy: value})}
                        options={['Yes', 'No']}
                      />
                    </div>
                  </section>
                </div>
              </div>
            </div>
          )}

          {/* Other Details Tab */}
          {activeTab === 'other-details' && (
            <div className="flex gap-6">
              {/* Left Sidebar Navigation */}
              <div className="w-64 flex-shrink-0">
                <div className="bg-white rounded-lg border border-gray-200 p-4 sticky top-6">
                  <nav className="space-y-1">
                    {[
                      { id: 'other-medical', label: 'Other Medical Details' },
                      { id: 'cervical-risk', label: 'Cervical Cancer Risk Factors' },
                      { id: 'other-info', label: 'Other Details' },
                      { id: 'breast-risk', label: 'Breast Cancer Risk Factors' }
                    ].map((section) => (
                      <a
                        key={section.id}
                        href={`#${section.id}`}
                        className="block px-3 py-2 rounded transition-colors"
                        style={{
                          fontSize: 'var(--text-sm)',
                          fontWeight: activeOtherDetailsSection === section.id ? 'var(--font-weight-semibold)' : 'var(--font-weight-normal)',
                          color: activeOtherDetailsSection === section.id ? 'var(--primary)' : '#111827',
                          backgroundColor: activeOtherDetailsSection === section.id ? '#F3F4F6' : 'transparent'
                        }}
                        onMouseEnter={(e) => {
                          if (activeOtherDetailsSection !== section.id) {
                            e.currentTarget.style.backgroundColor = '#F3F4F6';
                          }
                        }}
                        onMouseLeave={(e) => {
                          if (activeOtherDetailsSection !== section.id) {
                            e.currentTarget.style.backgroundColor = 'transparent';
                          }
                        }}
                      >
                        {section.label}
                      </a>
                    ))}
                  </nav>
                </div>
              </div>

              {/* Main Content Area */}
              <div className="flex-1 relative">
                {/* Sticky Edit/Save Buttons */}
                <div className="sticky bottom-0 z-10 py-4 flex justify-end gap-3">
                  {!isEditingOtherDetails ? (
                    <Button 
                      variant="outline" 
                      onClick={handleEditOtherDetails}
                      style={{
                        fontSize: 'var(--text-sm)',
                        fontWeight: 'var(--font-weight-medium)'
                      }}
                    >
                      Edit
                    </Button>
                  ) : (
                    <>
                      <Button 
                        variant="outline" 
                        onClick={handleCancelOtherDetails}
                        style={{
                          fontSize: 'var(--text-sm)',
                          fontWeight: 'var(--font-weight-medium)'
                        }}
                      >
                        Cancel
                      </Button>
                      <Button 
                        onClick={handleSaveOtherDetails}
                        style={{
                          fontSize: 'var(--text-sm)',
                          fontWeight: 'var(--font-weight-medium)',
                          backgroundColor: 'var(--primary)',
                          color: '#FFFFFF'
                        }}
                      >
                        Save
                      </Button>
                    </>
                  )}
                </div>

                <div className="space-y-6">
                  {/* Other Medical Details Section */}
                  <section id="other-medical" className="bg-white rounded-lg border border-gray-200 p-6">
                    <h2 className={FORM_CARD_SECTION_TITLE_CLASS}>
                      OTHER MEDICAL DETAILS
                    </h2>
                    
                    <div className="space-y-6">
                      <EditableTextArea
                        label="Other Medical/Surgical Illness"
                        value={otherDetailsFormData.otherMedicalIllness}
                        isEditing={isEditingOtherDetails}
                        onChange={(value) => setOtherDetailsFormData({...otherDetailsFormData, otherMedicalIllness: value})}
                        rows={5}
                      />
                      
                      <div className="grid grid-cols-2 gap-6">
                        <EditableField
                          label="Current Medication"
                          value={otherDetailsFormData.currentMedication}
                          isEditing={isEditingOtherDetails}
                          onChange={(value) => setOtherDetailsFormData({...otherDetailsFormData, currentMedication: value})}
                        />
                        
                        <EditableField
                          label="Any Drug Allergy"
                          value={otherDetailsFormData.drugAllergy}
                          isEditing={isEditingOtherDetails}
                          onChange={(value) => setOtherDetailsFormData({...otherDetailsFormData, drugAllergy: value})}
                        />
                      </div>
                      
                      <EditableField
                        label="High Risk Immunosuppressive clinical condition"
                        value={otherDetailsFormData.immunosuppressiveCondition}
                        isEditing={isEditingOtherDetails}
                        onChange={(value) => setOtherDetailsFormData({...otherDetailsFormData, immunosuppressiveCondition: value})}
                      />
                    </div>
                  </section>

                  {/* Cervical Cancer Risk Factors Section */}
                  <section id="cervical-risk" className="bg-white rounded-lg border border-gray-200 p-6">
                    <h2 className={FORM_CARD_SECTION_TITLE_CLASS}>
                      CERVICAL CANCER RISK FACTORS
                    </h2>
                    
                    <div className="grid grid-cols-2 gap-6">
                      <EditableField
                        label="Age at First Sexual Intercourse"
                        value={otherDetailsFormData.ageFirstIntercourse}
                        isEditing={isEditingOtherDetails}
                        onChange={(value) => setOtherDetailsFormData({...otherDetailsFormData, ageFirstIntercourse: value})}
                      />
                      
                      <EditableSelect
                        label="Multiple Sexual Partners"
                        value={otherDetailsFormData.multipleSexualPartners}
                        isEditing={isEditingOtherDetails}
                        onChange={(value) => setOtherDetailsFormData({...otherDetailsFormData, multipleSexualPartners: value})}
                        options={['Yes', 'No']}
                      />
                      
                      <EditableSelect
                        label="Smoking"
                        value={otherDetailsFormData.smoking}
                        isEditing={isEditingOtherDetails}
                        onChange={(value) => setOtherDetailsFormData({...otherDetailsFormData, smoking: value})}
                        options={['Yes', 'No']}
                      />
                      
                      <EditableField
                        label="Duration Smoking (in Yrs)"
                        value={otherDetailsFormData.smokingDuration}
                        isEditing={isEditingOtherDetails}
                        onChange={(value) => setOtherDetailsFormData({...otherDetailsFormData, smokingDuration: value})}
                      />
                      
                      <EditableSelect
                        label="History of Sexually Transmitted Infections"
                        value={otherDetailsFormData.sexuallyTransmittedInfections}
                        isEditing={isEditingOtherDetails}
                        onChange={(value) => setOtherDetailsFormData({...otherDetailsFormData, sexuallyTransmittedInfections: value})}
                        options={['Yes', 'No']}
                      />
                      
                      <EditableField
                        label="Use of OCP (in Yrs)"
                        value={otherDetailsFormData.ocpUse}
                        isEditing={isEditingOtherDetails}
                        onChange={(value) => setOtherDetailsFormData({...otherDetailsFormData, ocpUse: value})}
                      />
                      
                      <EditableSelect
                        label="HPV Vaccination"
                        value={otherDetailsFormData.hpvVaccinated}
                        isEditing={isEditingOtherDetails}
                        onChange={(value) => setOtherDetailsFormData({...otherDetailsFormData, hpvVaccinated: value})}
                        options={['Yes', 'No']}
                      />
                      
                      <div className="col-span-2">
                        <EditableTextArea
                          label="Other Details"
                          value={otherDetailsFormData.cervicalOtherDetails}
                          isEditing={isEditingOtherDetails}
                          onChange={(value) => setOtherDetailsFormData({...otherDetailsFormData, cervicalOtherDetails: value})}
                          rows={2}
                        />
                      </div>
                    </div>
                  </section>

                  {/* Other Details Section */}
                  <section id="other-info" className="bg-white rounded-lg border border-gray-200 p-6">
                    <h2 className={FORM_CARD_SECTION_TITLE_CLASS}>
                      OTHER DETAILS
                    </h2>
                    
                    <div className="grid grid-cols-2 gap-6">
                      <EditableField
                        label="Para (No. Of Children Delivered)"
                        value={otherDetailsFormData.childrenCount}
                        isEditing={isEditingOtherDetails}
                        onChange={(value) => setOtherDetailsFormData({...otherDetailsFormData, childrenCount: value})}
                      />
                      
                      <EditableField
                        label="Last Pap / HPV Test Done On"
                        value={otherDetailsFormData.lastPapTestDate}
                        isEditing={isEditingOtherDetails}
                        onChange={(value) => setOtherDetailsFormData({...otherDetailsFormData, lastPapTestDate: value})}
                        type="date"
                      />
                      
                      <EditableField
                        label="Last Pap / HPV Test Result"
                        value={otherDetailsFormData.lastPapTestResult}
                        isEditing={isEditingOtherDetails}
                        onChange={(value) => setOtherDetailsFormData({...otherDetailsFormData, lastPapTestResult: value})}
                      />
                      
                      <EditableField
                        label="Last Mammogram Date"
                        value={otherDetailsFormData.lastMammogramDate}
                        isEditing={isEditingOtherDetails}
                        onChange={(value) => setOtherDetailsFormData({...otherDetailsFormData, lastMammogramDate: value})}
                        type="date"
                      />
                      
                      <EditableSelect
                        label="CHAS Card Holder"
                        value={otherDetailsFormData.chasCardHolder}
                        isEditing={isEditingOtherDetails}
                        onChange={(value) => setOtherDetailsFormData({...otherDetailsFormData, chasCardHolder: value})}
                        options={['Yes', 'No']}
                      />
                    </div>
                  </section>

                  {/* Breast Cancer Risk Factors Section */}
                  <section id="breast-risk" className="bg-white rounded-lg border border-gray-200 p-6">
                    <h2 className={FORM_CARD_SECTION_TITLE_CLASS}>
                      BREAST CANCER RISK FACTORS
                    </h2>
                    
                    <div className="grid grid-cols-2 gap-6">
                      <EditableField
                        label="Age at Menarche (FMP)"
                        value={otherDetailsFormData.ageAtMenarche}
                        isEditing={isEditingOtherDetails}
                        onChange={(value) => setOtherDetailsFormData({...otherDetailsFormData, ageAtMenarche: value})}
                      />
                      
                      <EditableField
                        label="Age at Menopause"
                        value={otherDetailsFormData.ageAtMenopause}
                        isEditing={isEditingOtherDetails}
                        onChange={(value) => setOtherDetailsFormData({...otherDetailsFormData, ageAtMenopause: value})}
                      />
                      
                      <EditableField
                        label="Age at First Childbirth"
                        value={otherDetailsFormData.ageAtFirstChildbirth}
                        isEditing={isEditingOtherDetails}
                        onChange={(value) => setOtherDetailsFormData({...otherDetailsFormData, ageAtFirstChildbirth: value})}
                      />
                      
                      <EditableField
                        label="Duration of Use of HRT (in Yrs)"
                        value={otherDetailsFormData.hrtDuration}
                        isEditing={isEditingOtherDetails}
                        onChange={(value) => setOtherDetailsFormData({...otherDetailsFormData, hrtDuration: value})}
                      />
                      
                      <EditableSelect
                        label="Pre-Malignant Breast Conditions"
                        value={otherDetailsFormData.preMalignantConditions}
                        isEditing={isEditingOtherDetails}
                        onChange={(value) => setOtherDetailsFormData({...otherDetailsFormData, preMalignantConditions: value})}
                        options={['Yes', 'No']}
                      />
                      
                      <EditableTextArea
                        label="Breast Conditions (Details)"
                        value={otherDetailsFormData.breastConditionsDetails}
                        isEditing={isEditingOtherDetails}
                        onChange={(value) => setOtherDetailsFormData({...otherDetailsFormData, breastConditionsDetails: value})}
                        className="col-span-2"
                        rows={3}
                      />
                    </div>
                  </section>
                </div>
              </div>
            </div>
          )}

          {/* Screening Tab */}
          {activeTab === 'screening' && (
            <div className="space-y-6">
              <section id="screening-review" className="bg-white rounded-lg border border-gray-200 p-6">
                <h2 className={FORM_CARD_SECTION_TITLE_CLASS}>REVIEW</h2>
                <div className="grid grid-cols-1 md:grid-cols-2 gap-6">
                  <EditableSelect
                    label="Review period"
                    value={detailsFormData.reviewPeriod}
                    isEditing
                    onChange={(value) =>
                      setDetailsFormData({ ...detailsFormData, reviewPeriod: value })
                    }
                    options={[
                      { value: '6months', label: '6 months' },
                      { value: '1year', label: '1 year' },
                      { value: '3years', label: '3 years' },
                      { value: '5years', label: '5 years' },
                    ]}
                    emptyOptionLabel="-- Select --"
                  />
                  <EditableField
                    label="Next review date"
                    value={detailsFormData.nextReviewDate}
                    isEditing={false}
                    onChange={() => {}}
                  />
                </div>
              </section>
              <section id="screening-history" className="bg-white rounded-lg border border-gray-200 p-6">
                <h2 className={FORM_CARD_SECTION_TITLE_CLASS}>
                  SCREENING HISTORY
                </h2>
                
                <div className="overflow-x-auto">
                  <table className="w-full">
                    <thead style={{ backgroundColor: '#F8F9FA' }}>
                      <tr>
                        <th className="text-left py-3 px-4 border-b" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-semibold)', color: '#111827', borderColor: '#E5E7EB' }}>
                          Visit No.
                        </th>
                        <th className="text-left py-3 px-4 border-b" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-semibold)', color: '#111827', borderColor: '#E5E7EB' }}>
                          Date of Visit
                        </th>
                        <th className="text-left py-3 px-4 border-b" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-semibold)', color: '#111827', borderColor: '#E5E7EB' }}>
                          Doctor Name
                        </th>
                        <th className="text-left py-3 px-4 border-b" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-semibold)', color: '#111827', borderColor: '#E5E7EB' }}>
                          Test Type
                        </th>
                        <th className="text-left py-3 px-4 border-b" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-semibold)', color: '#111827', borderColor: '#E5E7EB' }}>
                          Result Date PAP
                        </th>
                        <th className="text-left py-3 px-4 border-b" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-semibold)', color: '#111827', borderColor: '#E5E7EB' }}>
                          Pap Result Class
                        </th>
                        <th className="text-left py-3 px-4 border-b" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-semibold)', color: '#111827', borderColor: '#E5E7EB' }}>
                          HPV 16
                        </th>
                        <th className="text-left py-3 px-4 border-b" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-semibold)', color: '#111827', borderColor: '#E5E7EB' }}>
                          HPV 18
                        </th>
                        <th className="text-left py-3 px-4 border-b" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-semibold)', color: '#111827', borderColor: '#E5E7EB' }}>
                          Other High Risk
                        </th>
                        <th className="text-left py-3 px-4 border-b" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-semibold)', color: '#111827', borderColor: '#E5E7EB' }}>
                          Next Appointment In
                        </th>
                        <th className="text-left py-3 px-4 border-b" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-semibold)', color: '#111827', borderColor: '#E5E7EB' }}>
                          Status
                        </th>
                      </tr>
                    </thead>
                    <tbody>
                      <tr className="border-b" style={{ borderColor: '#E5E7EB' }}>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          V001
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          15/11/2025
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          Dr. Sarah Chen
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          Pap Smear
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          20/11/2025
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          Negative
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          Negative
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          Negative
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          Negative
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          3 years
                        </td>
                        <td className="py-3 px-4">
                          <span className="px-3 py-1 rounded" style={{ 
                            backgroundColor: '#D1FAE5',
                            color: '#065F46',
                            fontSize: 'var(--text-xs)',
                            fontWeight: 'var(--font-weight-medium)'
                          }}>
                            Completed
                          </span>
                        </td>
                      </tr>
                      <tr className="border-b" style={{ borderColor: '#E5E7EB' }}>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          V002
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          20/05/2024
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          Dr. Michael Tan
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          Pap Smear + HPV Test
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          28/05/2024
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          Normal
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          Negative
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          Negative
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          Negative
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          1 year
                        </td>
                        <td className="py-3 px-4">
                          <span className="px-3 py-1 rounded" style={{ 
                            backgroundColor: '#D1FAE5',
                            color: '#065F46',
                            fontSize: 'var(--text-xs)',
                            fontWeight: 'var(--font-weight-medium)'
                          }}>
                            Completed
                          </span>
                        </td>
                      </tr>
                      <tr className="border-b" style={{ borderColor: '#E5E7EB' }}>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          V003
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          18/11/2023
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          Dr. Sarah Chen
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          Pap Smear
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          25/11/2023
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          Normal
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          -
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          -
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          -
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          6 months
                        </td>
                        <td className="py-3 px-4">
                          <span className="px-3 py-1 rounded" style={{ 
                            backgroundColor: '#D1FAE5',
                            color: '#065F46',
                            fontSize: 'var(--text-xs)',
                            fontWeight: 'var(--font-weight-medium)'
                          }}>
                            Completed
                          </span>
                        </td>
                      </tr>
                      <tr className="border-b" style={{ borderColor: '#E5E7EB' }}>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          V004
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          15/02/2023
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          Dr. Linda Wong
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          HPV Test
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          22/02/2023
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          -
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          Negative
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          Negative
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          Positive
                        </td>
                        <td className="py-3 px-4" style={{ fontSize: 'var(--text-sm)', fontWeight: 'var(--font-weight-normal)', color: '#272B30' }}>
                          3 months
                        </td>
                        <td className="py-3 px-4">
                          <span className="px-3 py-1 rounded" style={{ 
                            backgroundColor: '#FEF3C7',
                            color: '#92400E',
                            fontSize: 'var(--text-xs)',
                            fontWeight: 'var(--font-weight-medium)'
                          }}>
                            Follow-up Required
                          </span>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </section>
            </div>
          )}
        </div>
      </div>

      {/* Create Task Dialog */}
      <CreateTaskDialog
        open={isCreateTaskOpen}
        onOpenChange={setIsCreateTaskOpen}
        onCreateTask={handleCreateTask}
      />

      {/* Add Contact Log Dialog */}
      <AddContactLogDialog
        open={isAddContactLogOpen}
        onOpenChange={setIsAddContactLogOpen}
        onAddContactLog={handleAddContactLog}
      />

      {/* Add Note Dialog */}
      <AddNoteDialog
        open={isAddNoteOpen}
        onOpenChange={setIsAddNoteOpen}
        onSaveNote={handleAddNote}
      />

      {/* Manual Route Dialog */}
      <ManualRouteDialog
        open={isManualRouteOpen}
        onOpenChange={setIsManualRouteOpen}
        prospectName="John Tan"
      />

      {/* Add Tag Dialog */}
      <AddTagDialog
        open={isAddTagOpen}
        onOpenChange={setIsAddTagOpen}
        currentTags={prospectTags}
        onAddTags={(tags) => setProspectTags(tags)}
      />
    </div>
  );
}
