// scs-prospect-portal/src/components/referrals/ReferralDetails.tsx
import { ChevronLeft } from 'lucide-react';
import { Page } from '../../App';
import scsLogo from 'figma:asset/b0bcd35b322fc22141175a8b66dd339881b6f74b.png';
import { useState } from 'react';
import { Dialog, DialogContent, DialogDescription, DialogTitle } from '../ui/dialog';
import { FORM_CARD_SECTION_TITLE_CLASS } from '../../lib/formCardSectionTitle';
import { DdMmYyyyDateInput } from '../ui/dd-mm-yyyy-date-input';

interface ReferralDetailsProps {
  onNavigate: (page: Page) => void;
}

export function ReferralDetails({ onNavigate }: ReferralDetailsProps) {
  const [activeTab, setActiveTab] = useState<'personal' | 'appointment' | 'screening'>('personal');
  const [showAcceptDialog, setShowAcceptDialog] = useState(false);
  const [showWithdrawDialog, setShowWithdrawDialog] = useState(false);
  const [isCompleted, setIsCompleted] = useState(false);
  const [appointmentDate, setAppointmentDate] = useState('');
  const [treatmentEndDate, setTreatmentEndDate] = useState('');
  const [acceptTreatmentDate, setAcceptTreatmentDate] = useState('');

  return (
    <div className="h-screen flex flex-col" style={{ backgroundColor: '#F9FAFB' }}>
      {/* Header */}
      <div className="px-6 py-4" style={{ backgroundColor: '#FFFFFF', borderBottom: '1px solid var(--border)' }}>
        <div className="flex items-center">
          <img 
            src={scsLogo} 
            alt="Singapore Cancer Society" 
            style={{ height: '40px', cursor: 'pointer' }}
            onClick={() => onNavigate('segments')}
          />
        </div>
      </div>

      {/* Content */}
      <div className="flex-1 overflow-auto px-6 py-6">
        <div className="max-w-7xl mx-auto">
          {/* Title */}
          <div className="mb-6">
            <h1 style={{
              fontSize: 'var(--text-h3)',
              fontWeight: 'var(--font-weight-semibold)',
              color: 'var(--foreground)'
            }}>
              Referral Accepted: Update Progress
            </h1>
          </div>

          {/* Summary Section */}
          <div className="mb-6 grid grid-cols-5 gap-4">
            <div>
              <label style={{
                fontSize: 'var(--text-sm)',
                color: 'var(--foreground)',
                display: 'block',
                marginBottom: '4px',
                fontWeight: 'var(--font-weight-semibold)'
              }}>
                Case ID
              </label>
              <p style={{
                fontSize: 'var(--text-base)',
                color: 'var(--foreground)',
                fontWeight: 'var(--font-weight-normal)'
              }}>
                2025/CX/1124
              </p>
            </div>

            <div>
              <label style={{
                fontSize: 'var(--text-sm)',
                color: 'var(--foreground)',
                display: 'block',
                marginBottom: '4px',
                fontWeight: 'var(--font-weight-semibold)'
              }}>
                Screening Type
              </label>
              <p style={{
                fontSize: 'var(--text-base)',
                color: 'var(--foreground)',
                fontWeight: 'var(--font-weight-normal)'
              }}>
                HPV Test
              </p>
            </div>

            <div>
              <label style={{
                fontSize: 'var(--text-sm)',
                color: 'var(--foreground)',
                display: 'block',
                marginBottom: '4px',
                fontWeight: 'var(--font-weight-semibold)'
              }}>
                Referral to
              </label>
              <p style={{
                fontSize: 'var(--text-base)',
                color: 'var(--foreground)',
                fontWeight: 'var(--font-weight-normal)'
              }}>
                Gynaecologist
              </p>
            </div>

            <div>
              <label style={{
                fontSize: 'var(--text-sm)',
                color: 'var(--foreground)',
                display: 'block',
                marginBottom: '4px',
                fontWeight: 'var(--font-weight-semibold)'
              }}>
                Ref Submitted
              </label>
              <p style={{
                fontSize: 'var(--text-base)',
                color: 'var(--foreground)',
                fontWeight: 'var(--font-weight-normal)'
              }}>
                Wed, 17 Sep 2025, 09:00 AM
              </p>
            </div>

            <div>
              <label style={{
                fontSize: 'var(--text-sm)',
                color: 'var(--foreground)',
                display: 'block',
                marginBottom: '4px',
                fontWeight: 'var(--font-weight-semibold)'
              }}>
                Status
              </label>
              <p style={{
                fontSize: 'var(--text-base)',
                color: 'var(--foreground)',
                fontWeight: 'var(--font-weight-normal)'
              }}>
                Accepted
              </p>
            </div>
          </div>

          {/* Two Column Layout */}
          <div className="grid grid-cols-2 gap-6">
            {/* Left Column - Read Only Information */}
            <div className="space-y-6">
              {/* Personal Information Section */}
              <div className="p-6 rounded-lg" style={{ backgroundColor: '#FFFFFF', border: '1px solid var(--border)' }}>
                <h2 className={FORM_CARD_SECTION_TITLE_CLASS}>
                  Personal Information
                </h2>
                
                <div className="space-y-4">
                  <div className="grid grid-cols-2 gap-4">
                    <div>
                      <label style={{
                        fontSize: 'var(--text-sm)',
                        color: 'var(--muted-foreground)',
                        display: 'block',
                        marginBottom: '4px'
                      }}>
                        Name (as per NRIC)
                      </label>
                      <p style={{
                        fontSize: 'var(--text-base)',
                        color: 'var(--foreground)',
                        fontWeight: 'var(--font-weight-normal)'
                      }}>
                        Lim Zhi Hao
                      </p>
                    </div>

                    <div>
                      <label style={{
                        fontSize: 'var(--text-sm)',
                        color: 'var(--muted-foreground)',
                        display: 'block',
                        marginBottom: '4px'
                      }}>
                        NRIC/ FIN Number
                      </label>
                      <p style={{
                        fontSize: 'var(--text-base)',
                        color: 'var(--foreground)',
                        fontWeight: 'var(--font-weight-normal)'
                      }}>
                        S****567D ⊘
                      </p>
                    </div>
                  </div>

                  <div className="grid grid-cols-2 gap-4">
                    <div>
                      <label style={{
                        fontSize: 'var(--text-sm)',
                        color: 'var(--muted-foreground)',
                        display: 'block',
                        marginBottom: '4px'
                      }}>
                        Date of Birth
                      </label>
                      <p style={{
                        fontSize: 'var(--text-base)',
                        color: 'var(--foreground)',
                        fontWeight: 'var(--font-weight-normal)'
                      }}>
                        27 July 1956
                      </p>
                    </div>

                    <div>
                      <label style={{
                        fontSize: 'var(--text-sm)',
                        color: 'var(--muted-foreground)',
                        display: 'block',
                        marginBottom: '4px'
                      }}>
                        Gender
                      </label>
                      <p style={{
                        fontSize: 'var(--text-base)',
                        color: 'var(--foreground)',
                        fontWeight: 'var(--font-weight-normal)'
                      }}>
                        Female
                      </p>
                    </div>
                  </div>

                  <div className="grid grid-cols-2 gap-4">
                    <div>
                      <label style={{
                        fontSize: 'var(--text-sm)',
                        color: 'var(--muted-foreground)',
                        display: 'block',
                        marginBottom: '4px'
                      }}>
                        Residential Status
                      </label>
                      <p style={{
                        fontSize: 'var(--text-base)',
                        color: 'var(--foreground)',
                        fontWeight: 'var(--font-weight-normal)'
                      }}>
                        Singapore Citizen
                      </p>
                    </div>

                    <div>
                      <label style={{
                        fontSize: 'var(--text-sm)',
                        color: 'var(--muted-foreground)',
                        display: 'block',
                        marginBottom: '4px'
                      }}>
                        Spoken Language(s)
                      </label>
                      <p style={{
                        fontSize: 'var(--text-base)',
                        color: 'var(--foreground)',
                        fontWeight: 'var(--font-weight-normal)'
                      }}>
                        English, Chinese
                      </p>
                    </div>
                  </div>

                  <div className="grid grid-cols-2 gap-4">
                    <div>
                      <label style={{
                        fontSize: 'var(--text-sm)',
                        color: 'var(--muted-foreground)',
                        display: 'block',
                        marginBottom: '4px'
                      }}>
                        Mobile Number
                      </label>
                      <p style={{
                        fontSize: 'var(--text-base)',
                        color: 'var(--foreground)',
                        fontWeight: 'var(--font-weight-normal)'
                      }}>
                        +65 8971 2346
                      </p>
                    </div>

                    <div>
                      <label style={{
                        fontSize: 'var(--text-sm)',
                        color: 'var(--muted-foreground)',
                        display: 'block',
                        marginBottom: '4px'
                      }}>
                        Email
                      </label>
                      <p style={{
                        fontSize: 'var(--text-base)',
                        color: 'var(--foreground)',
                        fontWeight: 'var(--font-weight-normal)'
                      }}>
                        lim.jasmine94@gmail.com
                      </p>
                    </div>
                  </div>
                </div>
              </div>

              {/* Address Section */}
              <div className="p-6 rounded-lg" style={{ backgroundColor: '#FFFFFF', border: '1px solid var(--border)' }}>
                <h2 className={FORM_CARD_SECTION_TITLE_CLASS}>
                  Address
                </h2>
                
                <div className="space-y-4">
                  <div className="grid grid-cols-2 gap-4">
                    <div>
                      <label style={{
                        fontSize: 'var(--text-sm)',
                        color: 'var(--muted-foreground)',
                        display: 'block',
                        marginBottom: '4px'
                      }}>
                        Unit No.
                      </label>
                      <p style={{
                        fontSize: 'var(--text-base)',
                        color: 'var(--foreground)',
                        fontWeight: 'var(--font-weight-normal)'
                      }}>
                        345
                      </p>
                    </div>

                    <div>
                      <label style={{
                        fontSize: 'var(--text-sm)',
                        color: 'var(--muted-foreground)',
                        display: 'block',
                        marginBottom: '4px'
                      }}>
                        Postal Code
                      </label>
                      <p style={{
                        fontSize: 'var(--text-base)',
                        color: 'var(--foreground)',
                        fontWeight: 'var(--font-weight-normal)'
                      }}>
                        730991
                      </p>
                    </div>
                  </div>

                  <div>
                    <label style={{
                      fontSize: 'var(--text-sm)',
                      color: 'var(--muted-foreground)',
                      display: 'block',
                      marginBottom: '4px'
                    }}>
                      Street Name & Block No.
                    </label>
                    <p style={{
                      fontSize: 'var(--text-base)',
                      color: 'var(--foreground)',
                      fontWeight: 'var(--font-weight-normal)'
                    }}>
                      456 Tampines Avenue 3, Pinevale Condominium
                    </p>
                  </div>
                </div>
              </div>

              {/* Health Profile Section */}
              <div className="p-6 rounded-lg" style={{ backgroundColor: '#FFFFFF', border: '1px solid var(--border)' }}>
                <h2 className={FORM_CARD_SECTION_TITLE_CLASS}>
                  Health Profile
                </h2>
                
                <div className="space-y-4">
                  <div>
                    <label style={{
                      fontSize: 'var(--text-sm)',
                      color: 'var(--muted-foreground)',
                      display: 'block',
                      marginBottom: '4px'
                    }}>
                      Health Risk Factor
                    </label>
                    <p style={{
                      fontSize: 'var(--text-base)',
                      color: 'var(--foreground)',
                      fontWeight: 'var(--font-weight-normal)'
                    }}>
                      Smoking, Family history of diabetes
                    </p>
                  </div>

                  <div>
                    <label style={{
                      fontSize: 'var(--text-sm)',
                      color: 'var(--muted-foreground)',
                      display: 'block',
                      marginBottom: '4px'
                    }}>
                      Existing Conditions
                    </label>
                    <p style={{
                      fontSize: 'var(--text-base)',
                      color: 'var(--foreground)',
                      fontWeight: 'var(--font-weight-normal)'
                    }}>
                      Hypertension, Type 2 Diabetes
                    </p>
                  </div>

                  <div>
                    <label style={{
                      fontSize: 'var(--text-sm)',
                      color: 'var(--muted-foreground)',
                      display: 'block',
                      marginBottom: '4px'
                    }}>
                      Remarks
                    </label>
                    <p style={{
                      fontSize: 'var(--text-base)',
                      color: 'var(--foreground)',
                      fontWeight: 'var(--font-weight-normal)'
                    }}>
                      Remarks
                    </p>
                  </div>
                </div>
              </div>

              {/* Risk Assessment Section */}
              <div className="p-6 rounded-lg" style={{ backgroundColor: '#FFFFFF', border: '1px solid var(--border)' }}>
                <h2 className={FORM_CARD_SECTION_TITLE_CLASS}>
                  Risk Assessment
                </h2>
                
                <div className="space-y-4">
                  <div>
                    <label style={{
                      fontSize: 'var(--text-sm)',
                      color: 'var(--muted-foreground)',
                      display: 'block',
                      marginBottom: '4px'
                    }}>
                      Risk Level
                    </label>
                    <p style={{
                      fontSize: 'var(--text-base)',
                      color: 'var(--foreground)',
                      fontWeight: 'var(--font-weight-normal)'
                    }}>
                      High Risk
                    </p>
                  </div>

                  <div>
                    <label style={{
                      fontSize: 'var(--text-sm)',
                      color: 'var(--muted-foreground)',
                      display: 'block',
                      marginBottom: '4px'
                    }}>
                      Risk Factor
                    </label>
                    <p style={{
                      fontSize: 'var(--text-base)',
                      color: 'var(--foreground)',
                      fontWeight: 'var(--font-weight-normal)'
                    }}>
                      Family History, Age Factor, Lifestyle Risks
                    </p>
                  </div>
                </div>
              </div>
            </div>

            {/* Right Column - Update Progress Form */}
            <div>
              <div className="p-6 rounded-lg" style={{ backgroundColor: '#FFFFFF', border: '1px solid var(--border)' }}>
                <h2 className={FORM_CARD_SECTION_TITLE_CLASS}>
                  Update Progress
                </h2>

                <div className="space-y-4">
                  <div>
                    <label style={{
                      fontSize: 'var(--text-sm)',
                      color: 'var(--foreground)',
                      display: 'block',
                      marginBottom: '8px'
                    }}>
                      Appointment Date
                    </label>
                    <DdMmYyyyDateInput value={appointmentDate} onChange={setAppointmentDate} />
                  </div>

                  <div>
                    <label style={{
                      fontSize: 'var(--text-sm)',
                      color: 'var(--foreground)',
                      display: 'block',
                      marginBottom: '8px'
                    }}>
                      Clinic Name / Location
                    </label>
                    <input
                      type="text"
                      placeholder="Input Clinic Name / Location"
                      style={{
                        width: '100%',
                        padding: '10px 12px',
                        backgroundColor: '#FFFFFF',
                        border: '1px solid #CED4DA',
                        borderRadius: '6px',
                        fontSize: 'var(--text-base)',
                        color: '#4F575E'
                      }}
                    />
                  </div>

                  <div>
                    <label style={{
                      fontSize: 'var(--text-sm)',
                      color: 'var(--foreground)',
                      display: 'block',
                      marginBottom: '8px'
                    }}>
                      Outcome
                    </label>
                    <select
                      style={{
                        width: '100%',
                        padding: '10px 12px',
                        backgroundColor: '#FFFFFF',
                        border: '1px solid #CED4DA',
                        borderRadius: '6px',
                        fontSize: 'var(--text-base)',
                        color: '#4F575E',
                        appearance: 'none',
                        backgroundImage: `url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='12' height='12' viewBox='0 0 12 12'%3E%3Cpath fill='%234F575E' d='M6 9L1 4h10z'/%3E%3C/svg%3E")`,
                        backgroundRepeat: 'no-repeat',
                        backgroundPosition: 'right 12px center'
                      }}
                    >
                      <option>Select outcome</option>
                    </select>
                  </div>

                  <div>
                    <label style={{
                      fontSize: 'var(--text-sm)',
                      color: 'var(--foreground)',
                      display: 'block',
                      marginBottom: '8px'
                    }}>
                      Attending Doctor
                    </label>
                    <input
                      type="text"
                      placeholder="Input Attending Doctor"
                      style={{
                        width: '100%',
                        padding: '10px 12px',
                        backgroundColor: '#FFFFFF',
                        border: '1px solid #CED4DA',
                        borderRadius: '6px',
                        fontSize: 'var(--text-base)',
                        color: '#4F575E'
                      }}
                    />
                  </div>

                  <div>
                    <label style={{
                      fontSize: 'var(--text-sm)',
                      color: 'var(--foreground)',
                      display: 'block',
                      marginBottom: '8px'
                    }}>
                      Status
                    </label>
                    <select
                      style={{
                        width: '100%',
                        padding: '10px 12px',
                        backgroundColor: '#FFFFFF',
                        border: '1px solid #CED4DA',
                        borderRadius: '6px',
                        fontSize: 'var(--text-base)',
                        color: '#4F575E',
                        appearance: 'none',
                        backgroundImage: `url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='12' height='12' viewBox='0 0 12 12'%3E%3Cpath fill='%234F575E' d='M6 9L1 4h10z'/%3E%3C/svg%3E")`,
                        backgroundRepeat: 'no-repeat',
                        backgroundPosition: 'right 12px center'
                      }}
                    >
                      <option>Select status</option>
                    </select>
                  </div>

                  <div>
                    <div className="flex items-center gap-2 py-2">
                      <input
                        type="checkbox"
                        id="mark-completed"
                        checked={isCompleted}
                        onChange={() => setIsCompleted(!isCompleted)}
                        className="accent-white"
                        style={{
                          width: '16px',
                          height: '16px',
                          cursor: 'pointer',
                          accentColor: '#FFFFFF'
                        }}
                      />
                      <label
                        htmlFor="mark-completed"
                        style={{
                          fontSize: 'var(--text-base)',
                          color: 'var(--foreground)',
                          cursor: 'pointer'
                        }}
                      >
                        Mark treatment as completed
                      </label>
                    </div>

                    {/* Conditional fields when checkbox is checked */}
                    {isCompleted && (
                      <>
                        <div>
                          <label style={{
                            fontSize: 'var(--text-sm)',
                            color: 'var(--foreground)',
                            display: 'block',
                            marginBottom: '8px'
                          }}>
                            Treatment End Date
                          </label>
                          <DdMmYyyyDateInput value={treatmentEndDate} onChange={setTreatmentEndDate} />
                        </div>

                        <div>
                          <div className="flex items-center justify-between mb-2">
                            <label style={{
                              fontSize: 'var(--text-sm)',
                              color: 'var(--foreground)'
                            }}>
                              Outcome Summary*
                            </label>
                            <span style={{
                              fontSize: 'var(--text-sm)',
                              color: 'var(--muted-foreground)'
                            }}>
                              0/500
                            </span>
                          </div>
                          <textarea
                            placeholder="Add Outcome Summary"
                            rows={4}
                            style={{
                              width: '100%',
                              padding: '10px 12px',
                              backgroundColor: '#FFFFFF',
                              border: '1px solid #CED4DA',
                              borderRadius: '6px',
                              fontSize: 'var(--text-base)',
                              color: '#4F575E',
                              resize: 'vertical'
                            }}
                          />
                        </div>
                      </>
                    )}
                  </div>

                  <div className="flex justify-end pt-4">
                    <button
                      onClick={() => onNavigate('segments')}
                      style={{
                        padding: '10px 32px',
                        backgroundColor: '#7C51A1',
                        color: '#FFFFFF',
                        border: 'none',
                        borderRadius: '6px',
                        fontSize: 'var(--text-base)',
                        fontWeight: 'var(--font-weight-medium)',
                        cursor: 'pointer'
                      }}
                    >
                      Confirm
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>

      {/* Accept Dialog */}
      <Dialog open={showAcceptDialog} onOpenChange={setShowAcceptDialog}>
        <DialogContent style={{ maxWidth: '500px' }}>
          <DialogTitle style={{
            fontSize: 'var(--text-lg)',
            fontWeight: 'var(--font-weight-semibold)',
            color: 'var(--foreground)',
            marginBottom: '24px'
          }}>
            Accept & Start Treatment
          </DialogTitle>
          <DialogDescription style={{ display: 'none' }}>
            Accept referral and start treatment
          </DialogDescription>

          <div className="space-y-4">
            <div>
              <label style={{
                fontSize: 'var(--text-sm)',
                color: 'var(--foreground)',
                display: 'block',
                marginBottom: '8px'
              }}>
                Treatment Date*
              </label>
              <DdMmYyyyDateInput value={acceptTreatmentDate} onChange={setAcceptTreatmentDate} required />
            </div>

            <div>
              <label style={{
                fontSize: 'var(--text-sm)',
                color: 'var(--foreground)',
                display: 'block',
                marginBottom: '8px'
              }}>
                Treatment Location*
              </label>
              <input
                type="text"
                placeholder="Input Treatment Location"
                style={{
                  width: '100%',
                  padding: '10px 12px',
                  backgroundColor: '#FFFFFF',
                  border: '1px solid #CED4DA',
                  borderRadius: '6px',
                  fontSize: 'var(--text-base)',
                  color: '#4F575E'
                }}
              />
            </div>
          </div>

          <div className="flex justify-end gap-3 mt-6">
            <button
              onClick={() => setShowAcceptDialog(false)}
              style={{
                padding: '10px 24px',
                backgroundColor: '#FFFFFF',
                color: 'var(--foreground)',
                border: '1px solid var(--border)',
                borderRadius: '6px',
                fontSize: 'var(--text-base)',
                fontWeight: 'var(--font-weight-medium)',
                cursor: 'pointer'
              }}
            >
              Cancel
            </button>
            <button
              onClick={() => onNavigate('accept-referral')}
              style={{
                padding: '10px 24px',
                backgroundColor: '#7C51A1',
                color: '#FFFFFF',
                border: 'none',
                borderRadius: '6px',
                fontSize: 'var(--text-base)',
                fontWeight: 'var(--font-weight-medium)',
                cursor: 'pointer'
              }}
            >
              Confirm
            </button>
          </div>
        </DialogContent>
      </Dialog>

      {/* Withdraw Dialog */}
      <Dialog open={showWithdrawDialog} onOpenChange={setShowWithdrawDialog}>
        <DialogContent style={{ maxWidth: '500px' }}>
          <DialogTitle style={{
            fontSize: 'var(--text-lg)',
            fontWeight: 'var(--font-weight-semibold)',
            color: 'var(--foreground)',
            marginBottom: '24px'
          }}>
            Withdrawn Referral
          </DialogTitle>
          <DialogDescription style={{ display: 'none' }}>
            Withdraw the referral
          </DialogDescription>

          <div className="space-y-4">
            <div>
              <label style={{
                fontSize: 'var(--text-sm)',
                color: 'var(--foreground)',
                display: 'block',
                marginBottom: '8px'
              }}>
                Reason*
              </label>
              <select
                style={{
                  width: '100%',
                  padding: '10px 12px',
                  backgroundColor: '#FFFFFF',
                  border: '1px solid #CED4DA',
                  borderRadius: '6px',
                  fontSize: 'var(--text-base)',
                  color: '#4F575E',
                  appearance: 'none',
                  backgroundImage: `url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='12' height='12' viewBox='0 0 12 12'%3E%3Cpath fill='%234F575E' d='M6 9L1 4h10z'/%3E%3C/svg%3E")`,
                  backgroundRepeat: 'no-repeat',
                  backgroundPosition: 'right 12px center'
                }}
              >
                <option>Select Reason</option>
              </select>
            </div>

            <div>
              <div className="flex items-center justify-between mb-2">
                <label style={{
                  fontSize: 'var(--text-sm)',
                  color: 'var(--foreground)'
                }}>
                  Remarks
                </label>
                <span style={{
                  fontSize: 'var(--text-sm)',
                  color: 'var(--muted-foreground)'
                }}>
                  0/500
                </span>
              </div>
              <textarea
                placeholder="Add Remarks"
                rows={4}
                style={{
                  width: '100%',
                  padding: '10px 12px',
                  backgroundColor: '#FFFFFF',
                  border: '1px solid #CED4DA',
                  borderRadius: '6px',
                  fontSize: 'var(--text-base)',
                  color: '#4F575E',
                  resize: 'vertical'
                }}
              />
            </div>
          </div>

          <div className="flex justify-end gap-3 mt-6">
            <button
              onClick={() => setShowWithdrawDialog(false)}
              style={{
                padding: '10px 24px',
                backgroundColor: '#FFFFFF',
                color: 'var(--foreground)',
                border: '1px solid var(--border)',
                borderRadius: '6px',
                fontSize: 'var(--text-base)',
                fontWeight: 'var(--font-weight-medium)',
                cursor: 'pointer'
              }}
            >
              Cancel
            </button>
            <button
              onClick={() => onNavigate('withdrawn-referral')}
              style={{
                padding: '10px 24px',
                backgroundColor: '#7C51A1',
                color: '#FFFFFF',
                border: 'none',
                borderRadius: '6px',
                fontSize: 'var(--text-base)',
                fontWeight: 'var(--font-weight-medium)',
                cursor: 'pointer'
              }}
            >
              Confirm
            </button>
          </div>
        </DialogContent>
      </Dialog>
    </div>
  );
}
